<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Web Server Manual | MNet Manual</title>
  <meta name="description" content="MNet: an R package and Shiny app for integrative analysis of metabolomic and transcriptomic data." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Web Server Manual | MNet Manual" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="MNet: an R package and Shiny app for integrative analysis of metabolomic and transcriptomic data." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Web Server Manual | MNet Manual" />
  
  <meta name="twitter:description" content="MNet: an R package and Shiny app for integrative analysis of metabolomic and transcriptomic data." />
  



<meta name="date" content="2024-11-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="applying-mnet-in-r.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> OVERVIEW</a></li>
<li class="chapter" data-level="2" data-path="web-server-manual.html"><a href="web-server-manual.html"><i class="fa fa-check"></i><b>2</b> Web Server Manual</a>
<ul>
<li class="chapter" data-level="2.1" data-path="web-server-manual.html"><a href="web-server-manual.html#metabolic-subnetwork"><i class="fa fa-check"></i><b>2.1</b> Metabolic Subnetwork</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="web-server-manual.html"><a href="web-server-manual.html#met-gene-subnetwork"><i class="fa fa-check"></i><b>2.1.1</b> Met-Gene Subnetwork</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="web-server-manual.html"><a href="web-server-manual.html#extended-pathway-analyses"><i class="fa fa-check"></i><b>2.2</b> Extended Pathway Analyses</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="web-server-manual.html"><a href="web-server-manual.html#epea"><i class="fa fa-check"></i><b>2.2.1</b> ePEA</a></li>
<li class="chapter" data-level="2.2.2" data-path="web-server-manual.html"><a href="web-server-manual.html#epda"><i class="fa fa-check"></i><b>2.2.2</b> ePDA</a></li>
<li class="chapter" data-level="2.2.3" data-path="web-server-manual.html"><a href="web-server-manual.html#esea"><i class="fa fa-check"></i><b>2.2.3</b> eSEA</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="web-server-manual.html"><a href="web-server-manual.html#construction-of-knowledgebase-dbmnet"><i class="fa fa-check"></i><b>2.3</b> Construction of Knowledgebase dbMNet</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="web-server-manual.html"><a href="web-server-manual.html#dbkegg"><i class="fa fa-check"></i><b>2.3.1</b> dbKEGG</a></li>
<li class="chapter" data-level="2.3.2" data-path="web-server-manual.html"><a href="web-server-manual.html#dbnet"><i class="fa fa-check"></i><b>2.3.2</b> dbNet</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="web-server-manual.html"><a href="web-server-manual.html#deployment"><i class="fa fa-check"></i><b>2.4</b> Deployment</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="web-server-manual.html"><a href="web-server-manual.html#installing-r-4.4.1-for-ubuntu-20"><i class="fa fa-check"></i><b>2.4.1</b> Installing R 4.4.1 for Ubuntu 20</a></li>
<li class="chapter" data-level="2.4.2" data-path="web-server-manual.html"><a href="web-server-manual.html#installing-r-packages"><i class="fa fa-check"></i><b>2.4.2</b> Installing R Packages</a></li>
<li class="chapter" data-level="2.4.3" data-path="web-server-manual.html"><a href="web-server-manual.html#installing-shiny-sever"><i class="fa fa-check"></i><b>2.4.3</b> Installing shiny-sever</a></li>
<li class="chapter" data-level="2.4.4" data-path="web-server-manual.html"><a href="web-server-manual.html#shiny-app-server-and-logs"><i class="fa fa-check"></i><b>2.4.4</b> Shiny App Server and Logs</a></li>
<li class="chapter" data-level="2.4.5" data-path="web-server-manual.html"><a href="web-server-manual.html#firewall-firewalld-or-ufw"><i class="fa fa-check"></i><b>2.4.5</b> Firewall: firewalld or ufw</a></li>
<li class="chapter" data-level="2.4.6" data-path="web-server-manual.html"><a href="web-server-manual.html#apache2-config"><i class="fa fa-check"></i><b>2.4.6</b> Apache2 Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="applying-mnet-in-r.html"><a href="applying-mnet-in-r.html"><i class="fa fa-check"></i><b>3</b> Applying MNet in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="applying-mnet-in-r.html"><a href="applying-mnet-in-r.html#acute-myeloid-leukemia"><i class="fa fa-check"></i><b>3.1</b> Acute myeloid leukemia</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="applying-mnet-in-r.html"><a href="applying-mnet-in-r.html#input-data"><i class="fa fa-check"></i><b>3.1.1</b> Input data</a></li>
<li class="chapter" data-level="3.1.2" data-path="applying-mnet-in-r.html"><a href="applying-mnet-in-r.html#met-gene-subnetwork-1"><i class="fa fa-check"></i><b>3.1.2</b> Met-Gene Subnetwork</a></li>
<li class="chapter" data-level="3.1.3" data-path="applying-mnet-in-r.html"><a href="applying-mnet-in-r.html#extended-pathway-analyses-1"><i class="fa fa-check"></i><b>3.1.3</b> Extended Pathway Analyses</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="applying-mnet-in-r.html"><a href="applying-mnet-in-r.html#triple-negative-breast-cancer"><i class="fa fa-check"></i><b>3.2</b> Triple-negative breast cancer</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="applying-mnet-in-r.html"><a href="applying-mnet-in-r.html#met-gene-subnetwork-2"><i class="fa fa-check"></i><b>3.2.1</b> Met-Gene Subnetwork</a></li>
<li class="chapter" data-level="3.2.2" data-path="applying-mnet-in-r.html"><a href="applying-mnet-in-r.html#extended-pathway-analyses-2"><i class="fa fa-check"></i><b>3.2.2</b> Extended Pathway Analyses</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="applying-mnet-in-r.html"><a href="applying-mnet-in-r.html#spatial-metabolomics"><i class="fa fa-check"></i><b>3.3</b> Spatial metabolomics</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="applying-mnet-in-r.html"><a href="applying-mnet-in-r.html#pea"><i class="fa fa-check"></i><b>3.3.1</b> PEA</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MNet Manual</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="web-server-manual" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Web Server Manual<a href="web-server-manual.html#web-server-manual" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="metabolic-subnetwork" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Metabolic Subnetwork<a href="web-server-manual.html#metabolic-subnetwork" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>The metabolism-related subnetwork analysis is executed through an analyzer specifically designed to identify subnetwork based on input gene- and metabolite-level summary data.</strong></p>
<div id="met-gene-subnetwork" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Met-Gene Subnetwork<a href="web-server-manual.html#met-gene-subnetwork" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="interface" class="section level4 hasAnchor" number="2.1.1.1">
<h4><span class="header-section-number">2.1.1.1</span> Interface<a href="web-server-manual.html#interface" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<blockquote>
<strong>Procedure</strong>
</blockquote>
<ul>
<li>
<p>Step 1: Enter <strong>Metabolite Data</strong>, <strong>GeneExp Data</strong> and <strong>Group Data</strong>, respectively.</p>
<li>
<p>Step 2: Select <strong>Nodes Number</strong>.</p>
<p>In R network analysis, nodes number refers to the total count of nodes, representing individual elements or entities within the network.</p>
<li>
<p>Step 3: Select <strong>Figure Format</strong> and adjust <strong>figure width, height and DPI</strong>.</p>
<li>
<p>Step 4: Click the <strong>User</strong> panel to <strong>view the input and output</strong>, and finally click <strong>Figure Download</strong> and export the analysis results.</p>
</ul>
<p><img src="figure/1.M-G.png" width="1206" style="display: block; margin: auto;" /></p>
<blockquote>
<strong>Demo data</strong>
</blockquote>
<ul>
<li>
<p><strong>Expand the Demo Panel and click Metabolic Data to download demo data</strong>, which comprises an integrated analysis of metabolomic and transcriptomic profiles in triple-negative breast cancer.</p>
<ul>
<li>
<p><strong>Metabolite Data</strong>: an interactive table for user-input metabolic data with rows corresponding to metabolites and columns corresponding to samples.</p>
<p><img src="figure/Metabolite.png" width="815" /></p>
<li>
<p><strong>GeneExp Data</strong>: an interactive table for user-input metabolic data with rows corresponding to genes and columns correspond to the samples.</p>
<p><img src="figure/GeneExp.png" width="807" /></p>
<li>
<p><strong>Group Data</strong>: Group information.</p>
<p><img src="figure/GroupInfo.png" width="786" style="display: block; margin: auto auto auto 0;" /></p>
</ul>
</ul>
</div>
<div id="results" class="section level4 hasAnchor" number="2.1.1.2">
<h4><span class="header-section-number">2.1.1.2</span> Results<a href="web-server-manual.html#results" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>
<p>By utilising the dbNet knowledgebase and employing the dnet algorithm, MNet analyses a list of genes and metabolites along with their significance information, allowing a graphical display of the metabolism-related subnetwork that contains both genes and metabolites.</p>
</ul>
<img src="figure/subnetwork.png" width="560" style="display: block; margin: auto;" />
<center>
<b>Figure 1.</b> Visualization of the identified optimal subnetwork that best explains the biological processes comparing two groups. The colors represent the logFC (logarithm of fold change) of genes, with red and green indicating different expression levels, while yellow and blue represent the logFC of metabolites, indicating varying levels.
</center>
</div>
</div>
</div>
<div id="extended-pathway-analyses" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Extended Pathway Analyses<a href="web-server-manual.html#extended-pathway-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="epea" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> ePEA<a href="web-server-manual.html#epea" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Extended pathway enrichment analysis</p>
<div id="interface-1" class="section level4 hasAnchor" number="2.2.1.1">
<h4><span class="header-section-number">2.2.1.1</span> Interface<a href="web-server-manual.html#interface-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<blockquote>
<strong>Procedure</strong>
</blockquote>
<ul>
<li>
<p>Step 1: Enter <strong>Metabolite Data</strong>, <strong>GeneExp Data</strong> and <strong>Group Data</strong>, respectively.</p>
<li>
<p>Step 2: Select <strong>Log(FoldChange), Padjust Cutoff and Pathway Pcutoff</strong>, respectively.</p>
<ul>
<li>
<p><strong>Fold change</strong>: Identifies key metabolites with significant expression shifts between conditions, revealing potential metabolic alterations and pathway involvement in biological processes.</p>
<li>
<p><strong>Padjust Cutoff</strong>: Helps filter significant results by controlling for false positives, ensuring that only statistically robust pathways are identified for further investigation.</p>
<li>
<p><strong>Pathway Pcutoff</strong>: Sets a significance threshold, helping to identify pathways with meaningful changes while reducing the likelihood of false-positive findings.</p>
</ul>
<li>
<p>Step 3: Select <strong>Figure Format</strong> and adjust <strong>figure width, height and DPI</strong>.</p>
<li>
<p>Step 4: Click the <strong>User</strong> panel to <strong>view the input and output</strong>, and finally click <strong>Figure Download</strong> and export the analysis results.</p>
</ul>
<p><img src="figure/1.M-G.png" width="1206" style="display: block; margin: auto;" /></p>
<blockquote>
<strong>Demo data</strong>
</blockquote>
<ul>
<li>
<p><strong>Expand the Demo Panel and click Metabolic Data to download demo data</strong>, which comprises an integrated analysis of metabolomic and transcriptomic profiles in triple-negative breast cancer.</p>
<ul>
<li>
<p><strong>Metabolite Data</strong>: an interactive table for user-input metabolic data with rows corresponding to metabolites and columns corresponding to samples.</p>
<p><img src="figure/Metabolite.png" width="815" /></p>
<li>
<p><strong>GeneExp Data</strong>: an interactive table for user-input metabolic data with rows corresponding to genes and columns correspond to the samples.</p>
<p><img src="figure/GeneExp.png" width="807" /></p>
<li>
<p><strong>Group Data</strong>: Group information.</p>
<p><img src="figure/GroupInfo.png" width="786" style="display: block; margin: auto auto auto 0;" /></p>
</ul>
</ul>
</div>
<div id="results-1" class="section level4 hasAnchor" number="2.2.1.2">
<h4><span class="header-section-number">2.2.1.2</span> Results<a href="web-server-manual.html#results-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<img src="figure/2.ePEA.png" width="2550" style="display: block; margin: auto;" />
<center>
<b>Figure 1.</b> Extended pathway enrichment analysis. (A) Barplot of up-regulated metabolic pathways corresponding to metabolites and genes. (B) Dotplot of up-regulated metabolic pathways corresponding to metabolites and genes. (C) Barplot of down-regulated metabolic pathways corresponding to metabolites and genes. (D) Dotplot of down-regulated metabolic pathways corresponding to metabolites and genes.
</center>
</div>
</div>
<div id="epda" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> ePDA<a href="web-server-manual.html#epda" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Extended pathway differential abundance (ePDA) score</p>
<div id="interface-2" class="section level4 hasAnchor" number="2.2.2.1">
<h4><span class="header-section-number">2.2.2.1</span> Interface<a href="web-server-manual.html#interface-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<blockquote>
<strong>Procedure</strong>
</blockquote>
<ul>
<li>
<p>Step 1: Enter <strong>Metabolite Data</strong>, <strong>GeneExp Data</strong> and <strong>Group Data</strong>, respectively.</p>
<li>
<p>Step 2: Select <strong>Log(FoldChange), Padjust Cutoff and Pathway Pcutoff</strong>, respectively.</p>
<ul>
<li>
<p><strong>Fold change</strong>: Identifies key metabolites with significant expression shifts between conditions, revealing potential metabolic alterations and pathway involvement in biological processes.</p>
<li>
<p><strong>Padjust Cutoff</strong>: Helps filter significant results by controlling for false positives, ensuring that only statistically robust pathways are identified for further investigation.</p>
<li>
<p><strong>Pathway Pcutoff</strong>: Sets a significance threshold, helping to identify pathways with meaningful changes while reducing the likelihood of false-positive findings.</p>
</ul>
<li>
<p>Step 3: Select <strong>Figure Format</strong> and adjust <strong>figure width, height and DPI</strong>.</p>
<li>
<p>Step 4: Click the <strong>User</strong> panel to <strong>view the input and output</strong>, and finally click <strong>Figure Download</strong> and export the analysis results.</p>
</ul>
<p><img src="figure/1.M-G.png" width="1206" style="display: block; margin: auto;" /></p>
<blockquote>
<strong>Demo data</strong>
</blockquote>
<ul>
<li>
<p><strong>Expand the Demo Panel and click Metabolic Data to download demo data</strong>, which comprises an integrated analysis of metabolomic and transcriptomic profiles in triple-negative breast cancer.</p>
<ul>
<li>
<p><strong>Metabolite Data</strong>: an interactive table for user-input metabolic data with rows corresponding to metabolites and columns corresponding to samples.</p>
<p><img src="figure/Metabolite.png" width="815" /></p>
<li>
<p><strong>GeneExp Data</strong>: an interactive table for user-input metabolic data with rows corresponding to genes and columns correspond to the samples.</p>
<p><img src="figure/GeneExp.png" width="807" /></p>
<li>
<p><strong>Group Data</strong>: Group information.</p>
<p><img src="figure/GroupInfo.png" width="786" style="display: block; margin: auto auto auto 0;" /></p>
</ul>
</ul>
</div>
<div id="results-2" class="section level4 hasAnchor" number="2.2.2.2">
<h4><span class="header-section-number">2.2.2.2</span> Results<a href="web-server-manual.html#results-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<img src="figure/2.ePDA.png" width="70%" style="display: block; margin: auto;" />
<center>
<b>Figure 1.</b> ePDA score captures the tendency for a pathway to exhibit increased or decreased levels of genes and metabolites that are statistically significant differences between two group.
</center>
</div>
</div>
<div id="esea" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> eSEA<a href="web-server-manual.html#esea" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Extended pathway set enrichment analysis</p>
<div id="interface-3" class="section level4 hasAnchor" number="2.2.3.1">
<h4><span class="header-section-number">2.2.3.1</span> Interface<a href="web-server-manual.html#interface-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<blockquote>
<strong>Procedure</strong>
</blockquote>
<ul>
<li>
<p>Step 1: Enter <strong>Metabolite Data</strong>, <strong>GeneExp Data</strong> and <strong>Group Data</strong>, respectively.</p>
<li>
<p>Step 2: Select <strong>Log(FoldChange), Padjust Cutoff and Pathway Pcutoff</strong>, respectively.</p>
<ul>
<li>
<p><strong>Fold change</strong>: Identifies key metabolites with significant expression shifts between conditions, revealing potential metabolic alterations and pathway involvement in biological processes.</p>
<li>
<p><strong>Padjust Cutoff</strong>: Helps filter significant results by controlling for false positives, ensuring that only statistically robust pathways are identified for further investigation.</p>
<li>
<p><strong>Pathway Pcutoff</strong>: Sets a significance threshold, helping to identify pathways with meaningful changes while reducing the likelihood of false-positive findings.</p>
</ul>
<li>
<p>Step 3: Select <strong>Figure Format</strong> and adjust <strong>figure width, height and DPI</strong>.</p>
<li>
<p>Step 4: Click the <strong>User</strong> panel to <strong>view the input and output</strong>, and finally click <strong>Figure Download</strong> and export the analysis results.</p>
</ul>
<p><img src="figure/1.M-G.png" width="1206" style="display: block; margin: auto;" /></p>
<blockquote>
<strong>Demo data</strong>
</blockquote>
<ul>
<li>
<p><strong>Expand the Demo Panel and click Metabolic Data to download demo data</strong>, which comprises an integrated analysis of metabolomic and transcriptomic profiles in triple-negative breast cancer.</p>
<ul>
<li>
<p><strong>Metabolite Data</strong>: an interactive table for user-input metabolic data with rows corresponding to metabolites and columns corresponding to samples.</p>
<p><img src="figure/Metabolite.png" width="815" /></p>
<li>
<p><strong>GeneExp Data</strong>: an interactive table for user-input metabolic data with rows corresponding to genes and columns correspond to the samples.</p>
<p><img src="figure/GeneExp.png" width="807" /></p>
<li>
<p><strong>Group Data</strong>: Group information.</p>
<p><img src="figure/GroupInfo.png" width="786" style="display: block; margin: auto auto auto 0;" /></p>
</ul>
</ul>
</div>
<div id="results-3" class="section level4 hasAnchor" number="2.2.3.2">
<h4><span class="header-section-number">2.2.3.2</span> Results<a href="web-server-manual.html#results-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<img src="figure/2.eSEA.png" width="60%" style="display: block; margin: auto;" />
<center>
<b>Figure 1.</b> Extended pathway set enrichment analysis.
</center>
</div>
</div>
</div>
<div id="construction-of-knowledgebase-dbmnet" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Construction of Knowledgebase dbMNet<a href="web-server-manual.html#construction-of-knowledgebase-dbmnet" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Knowledgebase - dbMNet: dbKEGG for extended pathway analysis and dbNet for metabolism-related subnetwork analysis
The construction may takes more than 60 hours.</p>
<div id="dbkegg" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> dbKEGG<a href="web-server-manual.html#dbkegg" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>dbKEGG for extended pathway analysis</p>
<div id="kegg-pathways-metabolite-and-gene" class="section level4 hasAnchor" number="2.3.1.1">
<h4><span class="header-section-number">2.3.1.1</span> KEGG pathway’s metabolite and gene<a href="web-server-manual.html#kegg-pathways-metabolite-and-gene" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This is the metabolites and the genes in every metabolism KEGG pathway.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="web-server-manual.html#cb2-1" tabindex="-1"></a><span class="fu">library</span>(KEGGREST)</span>
<span id="cb2-2"><a href="web-server-manual.html#cb2-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-3"><a href="web-server-manual.html#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="web-server-manual.html#cb2-4" tabindex="-1"></a><span class="do">## All human metabolic pathways</span></span>
<span id="cb2-5"><a href="web-server-manual.html#cb2-5" tabindex="-1"></a>pathway_meta <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;input/pathway_hsa.txt&quot;</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header=</span>F) <span class="sc">%&gt;%</span></span>
<span id="cb2-6"><a href="web-server-manual.html#cb2-6" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb2-7"><a href="web-server-manual.html#cb2-7" tabindex="-1"></a></span>
<span id="cb2-8"><a href="web-server-manual.html#cb2-8" tabindex="-1"></a><span class="do">## Extract corresponding genes and metabolites for each pathway</span></span>
<span id="cb2-9"><a href="web-server-manual.html#cb2-9" tabindex="-1"></a>result_gene <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb2-10"><a href="web-server-manual.html#cb2-10" tabindex="-1"></a>result_compound <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb2-11"><a href="web-server-manual.html#cb2-11" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pathway_meta<span class="sc">$</span>V2)){</span>
<span id="cb2-12"><a href="web-server-manual.html#cb2-12" tabindex="-1"></a></span>
<span id="cb2-13"><a href="web-server-manual.html#cb2-13" tabindex="-1"></a>  <span class="fu">print</span>(pathway_meta<span class="sc">$</span>V2[i])</span>
<span id="cb2-14"><a href="web-server-manual.html#cb2-14" tabindex="-1"></a></span>
<span id="cb2-15"><a href="web-server-manual.html#cb2-15" tabindex="-1"></a>  path <span class="ot">&lt;-</span> <span class="fu">keggGet</span>(pathway_meta<span class="sc">$</span>V2[i])</span>
<span id="cb2-16"><a href="web-server-manual.html#cb2-16" tabindex="-1"></a></span>
<span id="cb2-17"><a href="web-server-manual.html#cb2-17" tabindex="-1"></a>  <span class="do">## Extract the genetic information of this pathway</span></span>
<span id="cb2-18"><a href="web-server-manual.html#cb2-18" tabindex="-1"></a>  gene.info <span class="ot">&lt;-</span> path[[<span class="dv">1</span>]]<span class="sc">$</span>GENE <span class="sc">%&gt;%</span></span>
<span id="cb2-19"><a href="web-server-manual.html#cb2-19" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-20"><a href="web-server-manual.html#cb2-20" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;V1&quot;</span><span class="ot">=</span><span class="st">&quot;.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-21"><a href="web-server-manual.html#cb2-21" tabindex="-1"></a>    tidyr<span class="sc">::</span><span class="fu">separate</span>(V1,<span class="at">sep=</span><span class="st">&quot;;&quot;</span>,<span class="st">&quot;V1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-22"><a href="web-server-manual.html#cb2-22" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">pull</span>(V1)</span>
<span id="cb2-23"><a href="web-server-manual.html#cb2-23" tabindex="-1"></a></span>
<span id="cb2-24"><a href="web-server-manual.html#cb2-24" tabindex="-1"></a>  <span class="do">## Extract gene symbols from genes</span></span>
<span id="cb2-25"><a href="web-server-manual.html#cb2-25" tabindex="-1"></a>  gene.symbol <span class="ot">&lt;-</span> <span class="fu">unique</span>(gene.info[<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(gene.info)<span class="sc">%%</span><span class="dv">2</span> <span class="sc">==</span> <span class="dv">0</span>])</span>
<span id="cb2-26"><a href="web-server-manual.html#cb2-26" tabindex="-1"></a>  <span class="co">#gene.id &lt;- gene.info[1:length(gene.info)%%2 == 1]</span></span>
<span id="cb2-27"><a href="web-server-manual.html#cb2-27" tabindex="-1"></a></span>
<span id="cb2-28"><a href="web-server-manual.html#cb2-28" tabindex="-1"></a>  <span class="do">## Generate a data frame matching gene symbol and Entrez ID</span></span>
<span id="cb2-29"><a href="web-server-manual.html#cb2-29" tabindex="-1"></a>  gene.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">type=</span><span class="st">&quot;gene&quot;</span>,<span class="at">name =</span> gene.symbol,<span class="at">kegg_pathwayid=</span>pathway_meta<span class="sc">$</span>V2[i],<span class="at">kegg_pathwayname=</span>path[[<span class="dv">1</span>]]<span class="sc">$</span>PATHWAY_MAP,<span class="at">kegg_category =</span>pathway_meta<span class="sc">$</span>V1[i])</span>
<span id="cb2-30"><a href="web-server-manual.html#cb2-30" tabindex="-1"></a>  result_gene <span class="ot">&lt;-</span> <span class="fu">rbind</span>(result_gene,gene.df)</span>
<span id="cb2-31"><a href="web-server-manual.html#cb2-31" tabindex="-1"></a></span>
<span id="cb2-32"><a href="web-server-manual.html#cb2-32" tabindex="-1"></a>  <span class="do">## Extract metabolite information for this pathway </span></span>
<span id="cb2-33"><a href="web-server-manual.html#cb2-33" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(path[[<span class="dv">1</span>]]<span class="sc">$</span>COMPOUND)<span class="sc">&gt;</span><span class="dv">0</span>) {</span>
<span id="cb2-34"><a href="web-server-manual.html#cb2-34" tabindex="-1"></a>    compound.info <span class="ot">&lt;-</span> path[[<span class="dv">1</span>]]<span class="sc">$</span>COMPOUND <span class="sc">%&gt;%</span></span>
<span id="cb2-35"><a href="web-server-manual.html#cb2-35" tabindex="-1"></a>      <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-36"><a href="web-server-manual.html#cb2-36" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;V1&quot;</span><span class="ot">=</span><span class="st">&quot;.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-37"><a href="web-server-manual.html#cb2-37" tabindex="-1"></a>      <span class="fu">rownames</span>()</span>
<span id="cb2-38"><a href="web-server-manual.html#cb2-38" tabindex="-1"></a></span>
<span id="cb2-39"><a href="web-server-manual.html#cb2-39" tabindex="-1"></a>    <span class="do">## Generate compound and corresponding pathway information</span></span>
<span id="cb2-40"><a href="web-server-manual.html#cb2-40" tabindex="-1"></a>    compound.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">type=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">name =</span> compound.info,<span class="at">kegg_pathwayid=</span>pathway_meta<span class="sc">$</span>V2[i],<span class="at">kegg_pathwayname=</span>path[[<span class="dv">1</span>]]<span class="sc">$</span>PATHWAY_MAP,<span class="at">kegg_category =</span>pathway_meta<span class="sc">$</span>V1[i])</span>
<span id="cb2-41"><a href="web-server-manual.html#cb2-41" tabindex="-1"></a>    result_compound <span class="ot">&lt;-</span> <span class="fu">rbind</span>(result_compound,compound.df)</span>
<span id="cb2-42"><a href="web-server-manual.html#cb2-42" tabindex="-1"></a>  }</span>
<span id="cb2-43"><a href="web-server-manual.html#cb2-43" tabindex="-1"></a></span>
<span id="cb2-44"><a href="web-server-manual.html#cb2-44" tabindex="-1"></a>}</span>
<span id="cb2-45"><a href="web-server-manual.html#cb2-45" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(result_gene,result_compound) <span class="sc">%&gt;%</span></span>
<span id="cb2-46"><a href="web-server-manual.html#cb2-46" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb2-47"><a href="web-server-manual.html#cb2-47" tabindex="-1"></a><span class="fu">dim</span>(result)</span>
<span id="cb2-48"><a href="web-server-manual.html#cb2-48" tabindex="-1"></a><span class="fu">write.table</span>(result,<span class="st">&quot;result/KEGGpathway_metabolite_gene.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="dbnet" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> dbNet<a href="web-server-manual.html#dbnet" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>dbNet for metabolism-related subnetwork analysis</p>
<div id="kegg-metabolite-metabolite-pairs-and-metabolite-gene-pairs" class="section level4 hasAnchor" number="2.3.2.1">
<h4><span class="header-section-number">2.3.2.1</span> KEGG metabolite-metabolite pairs and metabolite-gene pairs<a href="web-server-manual.html#kegg-metabolite-metabolite-pairs-and-metabolite-gene-pairs" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<blockquote>
<strong>KEGG’s reaction and compound’s download</strong>
</blockquote>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="web-server-manual.html#cb3-1" tabindex="-1"></a><span class="fu">library</span>(KEGGREST)</span>
<span id="cb3-2"><a href="web-server-manual.html#cb3-2" tabindex="-1"></a><span class="fu">library</span>(plyr)</span>
<span id="cb3-3"><a href="web-server-manual.html#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="web-server-manual.html#cb3-4" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;input/get.kegg.all.R&quot;</span>)</span>
<span id="cb3-5"><a href="web-server-manual.html#cb3-5" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;input/get.kegg.byId.R&quot;</span>)</span>
<span id="cb3-6"><a href="web-server-manual.html#cb3-6" tabindex="-1"></a></span>
<span id="cb3-7"><a href="web-server-manual.html#cb3-7" tabindex="-1"></a><span class="do">## Metabolic reactions and metabolite annotation information in the KEGG database</span></span>
<span id="cb3-8"><a href="web-server-manual.html#cb3-8" tabindex="-1"></a>keggAll <span class="ot">=</span> <span class="fu">get.kegg.all</span>()</span>
<span id="cb3-9"><a href="web-server-manual.html#cb3-9" tabindex="-1"></a></span>
<span id="cb3-10"><a href="web-server-manual.html#cb3-10" tabindex="-1"></a><span class="fu">saveRDS</span>(keggAll, <span class="at">file =</span> <span class="st">&quot;result/keggAll.RDS&quot;</span>)</span>
<span id="cb3-11"><a href="web-server-manual.html#cb3-11" tabindex="-1"></a></span>
<span id="cb3-12"><a href="web-server-manual.html#cb3-12" tabindex="-1"></a><span class="fu">dim</span>(keggAll<span class="sc">$</span>reaction)</span>
<span id="cb3-13"><a href="web-server-manual.html#cb3-13" tabindex="-1"></a><span class="fu">dim</span>(keggAll<span class="sc">$</span>compound)</span>
<span id="cb3-14"><a href="web-server-manual.html#cb3-14" tabindex="-1"></a></span>
<span id="cb3-15"><a href="web-server-manual.html#cb3-15" tabindex="-1"></a><span class="fu">write.csv</span>(keggAll<span class="sc">$</span>reaction,<span class="at">file =</span> <span class="st">&quot;result/keggAllreaction.csv&quot;</span>,<span class="at">row.names =</span> F)</span>
<span id="cb3-16"><a href="web-server-manual.html#cb3-16" tabindex="-1"></a><span class="fu">write.csv</span>(keggAll<span class="sc">$</span>compound,<span class="at">file =</span> <span class="st">&quot;result/keggAllcompound.csv&quot;</span>,<span class="at">row.names =</span> F)</span></code></pre></div>
<blockquote>
<strong>Downloading the enzyme and its corresponding gene information from the htext,and extract the gene and the enzyme’s relationship</strong>
</blockquote>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="web-server-manual.html#cb4-1" tabindex="-1"></a><span class="do">## Download the htext file.</span></span>
<span id="cb4-2"><a href="web-server-manual.html#cb4-2" tabindex="-1"></a>curl https<span class="sc">:</span><span class="er">//</span>www.genome.jp<span class="sc">/</span>kegg<span class="sc">-</span>bin<span class="sc">/</span>get_htext?hsa01000 <span class="sc">-</span>o result<span class="sc">/</span>hsa01000.keg</span>
<span id="cb4-3"><a href="web-server-manual.html#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="web-server-manual.html#cb4-4" tabindex="-1"></a>grep <span class="st">&quot;^E&quot;</span> result<span class="sc">/</span>hsa01000.keg <span class="sc">|</span> sed <span class="st">&#39;s/E        //g&#39;</span><span class="sc">|</span>cut <span class="sc">-</span>d<span class="st">&quot; &quot;</span> <span class="sc">-</span>f2 <span class="sc">|</span>cut <span class="sc">-</span>d<span class="st">&quot;;&quot;</span> <span class="sc">-</span>f1 <span class="sc">&gt;</span>result<span class="sc">/</span>gene.txt</span>
<span id="cb4-5"><a href="web-server-manual.html#cb4-5" tabindex="-1"></a></span>
<span id="cb4-6"><a href="web-server-manual.html#cb4-6" tabindex="-1"></a><span class="do">###sed &#39;s/\[EC\:/ /g&#39; is followed by a tab character</span></span>
<span id="cb4-7"><a href="web-server-manual.html#cb4-7" tabindex="-1"></a>grep <span class="st">&quot;^E&quot;</span> result<span class="sc">/</span>hsa01000.keg <span class="sc">|</span> sed <span class="st">&#39;s/E        //g&#39;</span><span class="sc">|</span>sed <span class="st">&#39;s/\[EC\:/ /g&#39;</span> <span class="sc">|</span>sed <span class="st">&#39;s/\] //g&#39;</span> <span class="sc">|</span>cut <span class="sc">-</span>f3 <span class="sc">&gt;</span>result<span class="sc">/</span>enzyme.txt</span>
<span id="cb4-8"><a href="web-server-manual.html#cb4-8" tabindex="-1"></a></span>
<span id="cb4-9"><a href="web-server-manual.html#cb4-9" tabindex="-1"></a>paste result<span class="sc">/</span>gene.txt result<span class="sc">/</span>enzyme.txt <span class="sc">|</span>grep <span class="sc">-</span>v <span class="st">&quot;D-dopachrome&quot;</span> <span class="sc">|</span>grep <span class="sc">-</span>v <span class="st">&quot;cytochrome&quot;</span><span class="sc">|</span>grep <span class="sc">-</span>v <span class="st">&quot;putative&quot;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>gene_enzyme.txt</span>
<span id="cb4-10"><a href="web-server-manual.html#cb4-10" tabindex="-1"></a></span>
<span id="cb4-11"><a href="web-server-manual.html#cb4-11" tabindex="-1"></a><span class="do">## The gene_enzyme.txt is like this:</span></span>
<span id="cb4-12"><a href="web-server-manual.html#cb4-12" tabindex="-1"></a>ADH5<span class="sc">^</span>I1.<span class="dv">1</span>.<span class="fl">1.284</span> <span class="dv">1</span>.<span class="dv">1</span>.<span class="fl">1.1</span><span class="sc">$</span></span>
<span id="cb4-13"><a href="web-server-manual.html#cb4-13" tabindex="-1"></a>ADH1A<span class="sc">^</span>I1.<span class="dv">1</span>.<span class="fl">1.1</span><span class="sc">$</span></span></code></pre></div>
<blockquote>
<strong>Extracting the metabolite-gene pairs and metabolite-metabolite pairs</strong>
</blockquote>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="web-server-manual.html#cb5-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb5-2"><a href="web-server-manual.html#cb5-2" tabindex="-1"></a></span>
<span id="cb5-3"><a href="web-server-manual.html#cb5-3" tabindex="-1"></a><span class="do">## Transformed the enzyme and its corresponding gene to one-to-one relationship</span></span>
<span id="cb5-4"><a href="web-server-manual.html#cb5-4" tabindex="-1"></a>gene_enzyme <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/gene_enzyme.txt&quot;</span>,<span class="at">header=</span>F) <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="web-server-manual.html#cb5-5" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-6"><a href="web-server-manual.html#cb5-6" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_rows</span>(<span class="st">&quot;V2&quot;</span>,<span class="at">sep=</span><span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-7"><a href="web-server-manual.html#cb5-7" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-8"><a href="web-server-manual.html#cb5-8" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;gene_name&quot;</span><span class="ot">=</span><span class="st">&quot;V1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-9"><a href="web-server-manual.html#cb5-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;enzyme&quot;</span><span class="ot">=</span><span class="st">&quot;V2&quot;</span>)</span>
<span id="cb5-10"><a href="web-server-manual.html#cb5-10" tabindex="-1"></a></span>
<span id="cb5-11"><a href="web-server-manual.html#cb5-11" tabindex="-1"></a><span class="do">## All the reaction in kegg</span></span>
<span id="cb5-12"><a href="web-server-manual.html#cb5-12" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;result/keggAllreaction.csv&quot;</span>)</span>
<span id="cb5-13"><a href="web-server-manual.html#cb5-13" tabindex="-1"></a></span>
<span id="cb5-14"><a href="web-server-manual.html#cb5-14" tabindex="-1"></a><span class="co"># Get all the metabolite and its corresponding gene from all the reaction in kegg</span></span>
<span id="cb5-15"><a href="web-server-manual.html#cb5-15" tabindex="-1"></a></span>
<span id="cb5-16"><a href="web-server-manual.html#cb5-16" tabindex="-1"></a><span class="do">## The metabolite-gene pairs are exacted from every compound in the equation and the gene with the enzyme</span></span>
<span id="cb5-17"><a href="web-server-manual.html#cb5-17" tabindex="-1"></a></span>
<span id="cb5-18"><a href="web-server-manual.html#cb5-18" tabindex="-1"></a>metabolite_gene <span class="ot">&lt;-</span> a <span class="sc">%&gt;%</span></span>
<span id="cb5-19"><a href="web-server-manual.html#cb5-19" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&quot;EQUATION&quot;</span>,<span class="st">&quot;ENZYME&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-20"><a href="web-server-manual.html#cb5-20" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_rows</span>(EQUATION,<span class="at">sep=</span><span class="st">&quot; &lt;=&gt; &quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-21"><a href="web-server-manual.html#cb5-21" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_rows</span>(EQUATION,<span class="at">sep=</span><span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-22"><a href="web-server-manual.html#cb5-22" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;^C&quot;</span>,EQUATION)) <span class="sc">%&gt;%</span></span>
<span id="cb5-23"><a href="web-server-manual.html#cb5-23" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_rows</span>(ENZYME,<span class="at">sep=</span><span class="st">&quot;///&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-24"><a href="web-server-manual.html#cb5-24" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;compound&quot;</span><span class="ot">=</span><span class="st">&quot;EQUATION&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-25"><a href="web-server-manual.html#cb5-25" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;enzyme&quot;</span><span class="ot">=</span><span class="st">&quot;ENZYME&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-26"><a href="web-server-manual.html#cb5-26" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene_enzyme,<span class="at">by=</span><span class="st">&quot;enzyme&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-27"><a href="web-server-manual.html#cb5-27" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_name)) <span class="sc">%&gt;%</span></span>
<span id="cb5-28"><a href="web-server-manual.html#cb5-28" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="st">&quot;enzyme&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-29"><a href="web-server-manual.html#cb5-29" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">compound=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[)]&quot;</span>,<span class="st">&quot;&quot;</span>,compound)) <span class="sc">%&gt;%</span></span>
<span id="cb5-30"><a href="web-server-manual.html#cb5-30" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">compound=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]m[)]&quot;</span>,<span class="st">&quot;&quot;</span>,compound)) <span class="sc">%&gt;%</span></span>
<span id="cb5-31"><a href="web-server-manual.html#cb5-31" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">compound=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[+]1[)]&quot;</span>,<span class="st">&quot;&quot;</span>,compound)) <span class="sc">%&gt;%</span></span>
<span id="cb5-32"><a href="web-server-manual.html#cb5-32" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">compound=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[+]m[)]&quot;</span>,<span class="st">&quot;&quot;</span>,compound)) <span class="sc">%&gt;%</span></span>
<span id="cb5-33"><a href="web-server-manual.html#cb5-33" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">compound=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[-]1[)]&quot;</span>,<span class="st">&quot;&quot;</span>,compound)) <span class="sc">%&gt;%</span></span>
<span id="cb5-34"><a href="web-server-manual.html#cb5-34" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-35"><a href="web-server-manual.html#cb5-35" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">src_type=</span><span class="st">&quot;metabolite&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-36"><a href="web-server-manual.html#cb5-36" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">dest_type=</span><span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-37"><a href="web-server-manual.html#cb5-37" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;src&quot;</span><span class="ot">=</span><span class="st">&quot;compound&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-38"><a href="web-server-manual.html#cb5-38" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;dest&quot;</span><span class="ot">=</span><span class="st">&quot;gene_name&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-39"><a href="web-server-manual.html#cb5-39" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&quot;src_type&quot;</span>,<span class="st">&quot;src&quot;</span>,<span class="st">&quot;dest_type&quot;</span>,<span class="st">&quot;dest&quot;</span>))</span>
<span id="cb5-40"><a href="web-server-manual.html#cb5-40" tabindex="-1"></a></span>
<span id="cb5-41"><a href="web-server-manual.html#cb5-41" tabindex="-1"></a><span class="co"># Get all the metabolite-metabolite pairs in the reactions</span></span>
<span id="cb5-42"><a href="web-server-manual.html#cb5-42" tabindex="-1"></a><span class="do">## Get the metabolite from the EQUATION, the metabolite-metabolite pairs are exacted from every compound in left of equation and every compound in right of equation.</span></span>
<span id="cb5-43"><a href="web-server-manual.html#cb5-43" tabindex="-1"></a>metabolite_metabolite <span class="ot">&lt;-</span> a <span class="sc">%&gt;%</span></span>
<span id="cb5-44"><a href="web-server-manual.html#cb5-44" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&quot;EQUATION&quot;</span>,<span class="st">&quot;ENZYME&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-45"><a href="web-server-manual.html#cb5-45" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate</span>(EQUATION,<span class="fu">c</span>(<span class="st">&quot;a&quot;</span>,<span class="st">&quot;b&quot;</span>),<span class="at">sep=</span><span class="st">&quot; &lt;=&gt; &quot;</span>)<span class="sc">%&gt;%</span></span>
<span id="cb5-46"><a href="web-server-manual.html#cb5-46" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_rows</span>(a,<span class="at">sep=</span><span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-47"><a href="web-server-manual.html#cb5-47" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_rows</span>(b,<span class="at">sep=</span><span class="st">&quot; &quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-48"><a href="web-server-manual.html#cb5-48" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;^C&quot;</span>,a)) <span class="sc">%&gt;%</span></span>
<span id="cb5-49"><a href="web-server-manual.html#cb5-49" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;^C&quot;</span>,b)) <span class="sc">%&gt;%</span></span>
<span id="cb5-50"><a href="web-server-manual.html#cb5-50" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&quot;a&quot;</span>,<span class="st">&quot;b&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-51"><a href="web-server-manual.html#cb5-51" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-52"><a href="web-server-manual.html#cb5-52" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_1=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[)]&quot;</span>,<span class="st">&quot;&quot;</span>,a)) <span class="sc">%&gt;%</span></span>
<span id="cb5-53"><a href="web-server-manual.html#cb5-53" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_1=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[+]m[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_1)) <span class="sc">%&gt;%</span></span>
<span id="cb5-54"><a href="web-server-manual.html#cb5-54" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_1=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]side&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_1)) <span class="sc">%&gt;%</span></span>
<span id="cb5-55"><a href="web-server-manual.html#cb5-55" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_1=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]m[+]n[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_1)) <span class="sc">%&gt;%</span></span>
<span id="cb5-56"><a href="web-server-manual.html#cb5-56" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_1=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[+]1[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_1)) <span class="sc">%&gt;%</span></span>
<span id="cb5-57"><a href="web-server-manual.html#cb5-57" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_1=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]m[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_1)) <span class="sc">%&gt;%</span></span>
<span id="cb5-58"><a href="web-server-manual.html#cb5-58" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_2=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[)]&quot;</span>,<span class="st">&quot;&quot;</span>,b)) <span class="sc">%&gt;%</span></span>
<span id="cb5-59"><a href="web-server-manual.html#cb5-59" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_2=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[+]m[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_2)) <span class="sc">%&gt;%</span></span>
<span id="cb5-60"><a href="web-server-manual.html#cb5-60" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_2=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]side&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_2)) <span class="sc">%&gt;%</span></span>
<span id="cb5-61"><a href="web-server-manual.html#cb5-61" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_2=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]m[+]n[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_2)) <span class="sc">%&gt;%</span></span>
<span id="cb5-62"><a href="web-server-manual.html#cb5-62" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_2=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[+]1[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_2)) <span class="sc">%&gt;%</span></span>
<span id="cb5-63"><a href="web-server-manual.html#cb5-63" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_2=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]m[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_2)) <span class="sc">%&gt;%</span></span>
<span id="cb5-64"><a href="web-server-manual.html#cb5-64" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_2=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[-]1[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_2)) <span class="sc">%&gt;%</span></span>
<span id="cb5-65"><a href="web-server-manual.html#cb5-65" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_2=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]x[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_2)) <span class="sc">%&gt;%</span></span>
<span id="cb5-66"><a href="web-server-manual.html#cb5-66" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_2=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[-]x[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_2)) <span class="sc">%&gt;%</span></span>
<span id="cb5-67"><a href="web-server-manual.html#cb5-67" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_2=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]n[+]2[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_2)) <span class="sc">%&gt;%</span></span>
<span id="cb5-68"><a href="web-server-manual.html#cb5-68" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">metabolite_2=</span><span class="fu">gsub</span>(<span class="st">&quot;[(]m[-]1[)]&quot;</span>,<span class="st">&quot;&quot;</span>,metabolite_2)) <span class="sc">%&gt;%</span></span>
<span id="cb5-69"><a href="web-server-manual.html#cb5-69" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">&quot;a&quot;</span>,<span class="st">&quot;b&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-70"><a href="web-server-manual.html#cb5-70" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(metabolite_1 <span class="sc">!=</span> metabolite_2) <span class="sc">%&gt;%</span></span>
<span id="cb5-71"><a href="web-server-manual.html#cb5-71" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">src_type=</span><span class="st">&quot;metabolite&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-72"><a href="web-server-manual.html#cb5-72" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">dest_type=</span><span class="st">&quot;metabolite&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-73"><a href="web-server-manual.html#cb5-73" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;src&quot;</span><span class="ot">=</span><span class="st">&quot;metabolite_1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-74"><a href="web-server-manual.html#cb5-74" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;dest&quot;</span><span class="ot">=</span><span class="st">&quot;metabolite_2&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-75"><a href="web-server-manual.html#cb5-75" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&quot;src_type&quot;</span>,<span class="st">&quot;src&quot;</span>,<span class="st">&quot;dest_type&quot;</span>,<span class="st">&quot;dest&quot;</span>))</span>
<span id="cb5-76"><a href="web-server-manual.html#cb5-76" tabindex="-1"></a></span>
<span id="cb5-77"><a href="web-server-manual.html#cb5-77" tabindex="-1"></a><span class="fu">write.table</span>(metabolite_gene,<span class="st">&quot;result/KEGG_metabolite_gene.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb5-78"><a href="web-server-manual.html#cb5-78" tabindex="-1"></a><span class="fu">write.table</span>(metabolite_metabolite,<span class="st">&quot;result/KEGG_metabolite_metabolite.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
</div>
<div id="graphite" class="section level4 hasAnchor" number="2.3.2.2">
<h4><span class="header-section-number">2.3.2.2</span> Graphite<a href="web-server-manual.html#graphite" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<blockquote>
<strong>Download all the data from graphite</strong>
</blockquote>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="web-server-manual.html#cb6-1" tabindex="-1"></a><span class="fu">library</span>(graphite)</span>
<span id="cb6-2"><a href="web-server-manual.html#cb6-2" tabindex="-1"></a><span class="fu">library</span>(clipper)</span>
<span id="cb6-3"><a href="web-server-manual.html#cb6-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb6-4"><a href="web-server-manual.html#cb6-4" tabindex="-1"></a></span>
<span id="cb6-5"><a href="web-server-manual.html#cb6-5" tabindex="-1"></a>kpaths <span class="ot">&lt;-</span> <span class="fu">pathways</span>(<span class="st">&quot;hsapiens&quot;</span>, <span class="st">&quot;kegg&quot;</span>)</span>
<span id="cb6-6"><a href="web-server-manual.html#cb6-6" tabindex="-1"></a>kpaths_result <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb6-7"><a href="web-server-manual.html#cb6-7" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(kpaths)) {</span>
<span id="cb6-8"><a href="web-server-manual.html#cb6-8" tabindex="-1"></a>  kid <span class="ot">&lt;-</span> <span class="fu">attributes</span>(kpaths[[i]])<span class="sc">$</span>id</span>
<span id="cb6-9"><a href="web-server-manual.html#cb6-9" tabindex="-1"></a>  ktitle <span class="ot">&lt;-</span> <span class="fu">attributes</span>(kpaths[[i]])<span class="sc">$</span>title</span>
<span id="cb6-10"><a href="web-server-manual.html#cb6-10" tabindex="-1"></a>  kpaths_1 <span class="ot">&lt;-</span> <span class="fu">convertIdentifiers</span>(<span class="fu">convertIdentifiers</span>(kpaths[[i]], <span class="st">&quot;symbol&quot;</span>),<span class="st">&quot;KEGGCOMP&quot;</span>)</span>
<span id="cb6-11"><a href="web-server-manual.html#cb6-11" tabindex="-1"></a>  kpaths_result_temp <span class="ot">&lt;-</span> <span class="fu">edges</span>(kpaths_1,<span class="st">&quot;mixed&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-12"><a href="web-server-manual.html#cb6-12" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathwayid=</span>kid) <span class="sc">%&gt;%</span></span>
<span id="cb6-13"><a href="web-server-manual.html#cb6-13" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway=</span>ktitle)</span>
<span id="cb6-14"><a href="web-server-manual.html#cb6-14" tabindex="-1"></a>  kpaths_result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(kpaths_result,kpaths_result_temp)</span>
<span id="cb6-15"><a href="web-server-manual.html#cb6-15" tabindex="-1"></a>}</span>
<span id="cb6-16"><a href="web-server-manual.html#cb6-16" tabindex="-1"></a><span class="fu">write.table</span>(kpaths_result,<span class="st">&quot;result/Graphite/gene-metabolite-kegg.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span>F)</span>
<span id="cb6-17"><a href="web-server-manual.html#cb6-17" tabindex="-1"></a></span>
<span id="cb6-18"><a href="web-server-manual.html#cb6-18" tabindex="-1"></a>spaths <span class="ot">&lt;-</span> <span class="fu">pathways</span>(<span class="st">&quot;hsapiens&quot;</span>, <span class="st">&quot;smpdb&quot;</span>)</span>
<span id="cb6-19"><a href="web-server-manual.html#cb6-19" tabindex="-1"></a>smpdb_result <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb6-20"><a href="web-server-manual.html#cb6-20" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(spaths)) {</span>
<span id="cb6-21"><a href="web-server-manual.html#cb6-21" tabindex="-1"></a>  kid <span class="ot">&lt;-</span> <span class="fu">attributes</span>(spaths[[i]])<span class="sc">$</span>id</span>
<span id="cb6-22"><a href="web-server-manual.html#cb6-22" tabindex="-1"></a>  ktitle <span class="ot">&lt;-</span> <span class="fu">attributes</span>(spaths[[i]])<span class="sc">$</span>title</span>
<span id="cb6-23"><a href="web-server-manual.html#cb6-23" tabindex="-1"></a>  smpdb_1 <span class="ot">&lt;-</span> <span class="fu">convertIdentifiers</span>(<span class="fu">convertIdentifiers</span>(spaths[[i]], <span class="st">&quot;symbol&quot;</span>),<span class="st">&quot;KEGGCOMP&quot;</span>)</span>
<span id="cb6-24"><a href="web-server-manual.html#cb6-24" tabindex="-1"></a>  smpdb_result_temp <span class="ot">&lt;-</span> <span class="fu">edges</span>(smpdb_1, <span class="st">&quot;mixed&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-25"><a href="web-server-manual.html#cb6-25" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathwayid=</span>kid) <span class="sc">%&gt;%</span></span>
<span id="cb6-26"><a href="web-server-manual.html#cb6-26" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway=</span>ktitle)</span>
<span id="cb6-27"><a href="web-server-manual.html#cb6-27" tabindex="-1"></a></span>
<span id="cb6-28"><a href="web-server-manual.html#cb6-28" tabindex="-1"></a>  smpdb_result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(smpdb_result,smpdb_result_temp)</span>
<span id="cb6-29"><a href="web-server-manual.html#cb6-29" tabindex="-1"></a>}</span>
<span id="cb6-30"><a href="web-server-manual.html#cb6-30" tabindex="-1"></a><span class="fu">write.table</span>(smpdb_result,<span class="st">&quot;result/Graphite/gene-metabolite-smpdb.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span>F)</span>
<span id="cb6-31"><a href="web-server-manual.html#cb6-31" tabindex="-1"></a></span>
<span id="cb6-32"><a href="web-server-manual.html#cb6-32" tabindex="-1"></a>wikipaths <span class="ot">&lt;-</span> <span class="fu">pathways</span>(<span class="st">&quot;hsapiens&quot;</span>, <span class="st">&quot;wikipathways&quot;</span>)</span>
<span id="cb6-33"><a href="web-server-manual.html#cb6-33" tabindex="-1"></a>wikipaths_result <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb6-34"><a href="web-server-manual.html#cb6-34" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(wikipaths)) {</span>
<span id="cb6-35"><a href="web-server-manual.html#cb6-35" tabindex="-1"></a>  kid <span class="ot">&lt;-</span> <span class="fu">attributes</span>(wikipaths[[i]])<span class="sc">$</span>id</span>
<span id="cb6-36"><a href="web-server-manual.html#cb6-36" tabindex="-1"></a>  ktitle <span class="ot">&lt;-</span> <span class="fu">attributes</span>(wikipaths[[i]])<span class="sc">$</span>title</span>
<span id="cb6-37"><a href="web-server-manual.html#cb6-37" tabindex="-1"></a>  wikipaths_1 <span class="ot">&lt;-</span> <span class="fu">convertIdentifiers</span>(<span class="fu">convertIdentifiers</span>(wikipaths[[i]], <span class="st">&quot;symbol&quot;</span>),<span class="st">&quot;KEGGCOMP&quot;</span>)</span>
<span id="cb6-38"><a href="web-server-manual.html#cb6-38" tabindex="-1"></a>  wikipaths_result_temp <span class="ot">&lt;-</span> <span class="fu">edges</span>(wikipaths_1,<span class="st">&quot;mixed&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-39"><a href="web-server-manual.html#cb6-39" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathwayid=</span>kid) <span class="sc">%&gt;%</span></span>
<span id="cb6-40"><a href="web-server-manual.html#cb6-40" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway=</span>ktitle)</span>
<span id="cb6-41"><a href="web-server-manual.html#cb6-41" tabindex="-1"></a>  wikipaths_result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(wikipaths_result,wikipaths_result_temp)</span>
<span id="cb6-42"><a href="web-server-manual.html#cb6-42" tabindex="-1"></a></span>
<span id="cb6-43"><a href="web-server-manual.html#cb6-43" tabindex="-1"></a>}</span>
<span id="cb6-44"><a href="web-server-manual.html#cb6-44" tabindex="-1"></a><span class="fu">write.table</span>(wikipaths_result,<span class="st">&quot;result/Graphite/gene-metabolite-wikipaths.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span>F)</span>
<span id="cb6-45"><a href="web-server-manual.html#cb6-45" tabindex="-1"></a></span>
<span id="cb6-46"><a href="web-server-manual.html#cb6-46" tabindex="-1"></a>reactomepaths <span class="ot">&lt;-</span> <span class="fu">pathways</span>(<span class="st">&quot;hsapiens&quot;</span>, <span class="st">&quot;reactome&quot;</span>)</span>
<span id="cb6-47"><a href="web-server-manual.html#cb6-47" tabindex="-1"></a>reactome_result <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb6-48"><a href="web-server-manual.html#cb6-48" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(reactomepaths)) {</span>
<span id="cb6-49"><a href="web-server-manual.html#cb6-49" tabindex="-1"></a>  kid <span class="ot">&lt;-</span> <span class="fu">attributes</span>(reactomepaths[[i]])<span class="sc">$</span>id</span>
<span id="cb6-50"><a href="web-server-manual.html#cb6-50" tabindex="-1"></a>  ktitle <span class="ot">&lt;-</span> <span class="fu">attributes</span>(reactomepaths[[i]])<span class="sc">$</span>title</span>
<span id="cb6-51"><a href="web-server-manual.html#cb6-51" tabindex="-1"></a>  reactome_1 <span class="ot">&lt;-</span> <span class="fu">convertIdentifiers</span>(<span class="fu">convertIdentifiers</span>(reactomepaths[[i]], <span class="st">&quot;symbol&quot;</span>),<span class="st">&quot;KEGGCOMP&quot;</span>)</span>
<span id="cb6-52"><a href="web-server-manual.html#cb6-52" tabindex="-1"></a>  reactome_result_temp <span class="ot">&lt;-</span> <span class="fu">edges</span>(reactome_1,<span class="st">&quot;mixed&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-53"><a href="web-server-manual.html#cb6-53" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathwayid=</span>kid) <span class="sc">%&gt;%</span></span>
<span id="cb6-54"><a href="web-server-manual.html#cb6-54" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway=</span>ktitle)</span>
<span id="cb6-55"><a href="web-server-manual.html#cb6-55" tabindex="-1"></a>  reactome_result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(reactome_result,reactome_result_temp)</span>
<span id="cb6-56"><a href="web-server-manual.html#cb6-56" tabindex="-1"></a>}</span>
<span id="cb6-57"><a href="web-server-manual.html#cb6-57" tabindex="-1"></a><span class="fu">write.table</span>(reactome_result,<span class="st">&quot;result/Graphite/gene-metabolite-reactome.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span>F)</span></code></pre></div>
<blockquote>
<strong>Download the information of metabolism pathways</strong>
</blockquote>
<p>Download all gpml files of homo-species from Wikipathway and extract the file names</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="web-server-manual.html#cb7-1" tabindex="-1"></a><span class="co">## Wikipathway</span></span>
<span id="cb7-2"><a href="web-server-manual.html#cb7-2" tabindex="-1"></a><span class="fu">wget</span> https://data.wikipathways.org/current/gpml/wikipathways-20241010-gpml-Homo_sapiens.zip</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="web-server-manual.html#cb8-1" tabindex="-1"></a>ls wikipathways<span class="dv">-20241010</span><span class="sc">-</span>gpml<span class="sc">-</span>Homo_sapiens<span class="sc">/</span><span class="er">*</span> <span class="er">&gt;</span>all.txt</span>
<span id="cb8-2"><a href="web-server-manual.html#cb8-2" tabindex="-1"></a>rev all.txt<span class="sc">|</span>cut <span class="sc">-</span>d<span class="st">&quot;_&quot;</span> <span class="sc">-</span>f2<span class="sc">|</span>rev <span class="sc">&gt;</span>all.<span class="fl">1.</span>txt</span>
<span id="cb8-3"><a href="web-server-manual.html#cb8-3" tabindex="-1"></a>cat <span class="sc">/</span>dev<span class="sc">/</span>null <span class="sc">&gt;</span> test.txt</span>
<span id="cb8-4"><a href="web-server-manual.html#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="web-server-manual.html#cb8-5" tabindex="-1"></a></span>
<span id="cb8-6"><a href="web-server-manual.html#cb8-6" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb8-7"><a href="web-server-manual.html#cb8-7" tabindex="-1"></a>cat all.<span class="fl">1.</span>txt <span class="sc">|</span><span class="cf">while</span> read line</span>
<span id="cb8-8"><a href="web-server-manual.html#cb8-8" tabindex="-1"></a>do</span>
<span id="cb8-9"><a href="web-server-manual.html#cb8-9" tabindex="-1"></a>grep TERM homo<span class="sc">/</span><span class="er">*$</span>{line}_<span class="sc">*</span>.gpml <span class="sc">|</span>cut <span class="sc">-</span>d<span class="st">&quot;&gt;&quot;</span> <span class="sc">-</span>f2<span class="sc">|</span>cut <span class="sc">-</span>d<span class="st">&quot;&lt;&quot;</span> <span class="sc">-</span>f1<span class="sc">|</span>  tr <span class="sc">-</span>s <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span> <span class="st">&quot;;&quot;</span><span class="sc">|</span>sed <span class="sc">-</span>e <span class="st">&#39;s/;$//g&#39;</span> <span class="sc">|</span> sed <span class="st">&quot;s/^/${line}</span></span>
<span id="cb8-10"><a href="web-server-manual.html#cb8-10" tabindex="-1"></a><span class="st">        /&quot;</span> <span class="sc">&gt;</span><span class="er">&gt;</span>test.txt</span>
<span id="cb8-11"><a href="web-server-manual.html#cb8-11" tabindex="-1"></a>done</span>
<span id="cb8-12"><a href="web-server-manual.html#cb8-12" tabindex="-1"></a></span>
<span id="cb8-13"><a href="web-server-manual.html#cb8-13" tabindex="-1"></a>grep <span class="sc">-</span>i metabolic test.txt <span class="sc">&gt;</span>test_metabolic.txt</span>
<span id="cb8-14"><a href="web-server-manual.html#cb8-14" tabindex="-1"></a>}</span>
<span id="cb8-15"><a href="web-server-manual.html#cb8-15" tabindex="-1"></a></span>
<span id="cb8-16"><a href="web-server-manual.html#cb8-16" tabindex="-1"></a>cat all.<span class="fl">1.</span>txt <span class="sc">|</span> <span class="cf">while</span> read line</span>
<span id="cb8-17"><a href="web-server-manual.html#cb8-17" tabindex="-1"></a>do</span>
<span id="cb8-18"><a href="web-server-manual.html#cb8-18" tabindex="-1"></a>grep TERM homo<span class="sc">/</span><span class="er">*$</span>{line}_<span class="sc">*</span>.gpml <span class="sc">|</span> cut <span class="sc">-</span>d<span class="st">&quot;&gt;&quot;</span> <span class="sc">-</span>f2 <span class="sc">|</span> cut <span class="sc">-</span>d<span class="st">&quot;&lt;&quot;</span> <span class="sc">-</span>f1 <span class="sc">|</span> tr <span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span> <span class="st">&#39;;&#39;</span> <span class="sc">|</span> sed <span class="sc">-</span>e <span class="st">&#39;s/;$//&#39;</span> <span class="sc">|</span> sed <span class="st">&quot;s/^/${line} /&quot;</span> <span class="sc">&gt;</span><span class="er">&gt;</span> test.txt</span>
<span id="cb8-19"><a href="web-server-manual.html#cb8-19" tabindex="-1"></a>done</span>
<span id="cb8-20"><a href="web-server-manual.html#cb8-20" tabindex="-1"></a>grep <span class="sc">-</span>i metabolic test.txt <span class="sc">&gt;</span>test_metabolic.txt</span>
<span id="cb8-21"><a href="web-server-manual.html#cb8-21" tabindex="-1"></a></span>
<span id="cb8-22"><a href="web-server-manual.html#cb8-22" tabindex="-1"></a><span class="do">## SMPDB</span></span>
<span id="cb8-23"><a href="web-server-manual.html#cb8-23" tabindex="-1"></a>wget https<span class="sc">:</span><span class="er">//</span>smpdb.ca<span class="sc">/</span>downloads<span class="sc">/</span>smpdb_pathways.csv.zip</span>
<span id="cb8-24"><a href="web-server-manual.html#cb8-24" tabindex="-1"></a></span>
<span id="cb8-25"><a href="web-server-manual.html#cb8-25" tabindex="-1"></a><span class="do">## Reactome</span></span>
<span id="cb8-26"><a href="web-server-manual.html#cb8-26" tabindex="-1"></a>R<span class="sc">-</span>HSA<span class="dv">-1430728</span> is the metabolism, then choose the hierarchical is lower than it</span>
<span id="cb8-27"><a href="web-server-manual.html#cb8-27" tabindex="-1"></a>wget https<span class="sc">:</span><span class="er">//</span>reactome.org<span class="sc">/</span>download<span class="sc">/</span>current<span class="sc">/</span>ReactomePathwaysRelation.txt</span>
<span id="cb8-28"><a href="web-server-manual.html#cb8-28" tabindex="-1"></a></span>
<span id="cb8-29"><a href="web-server-manual.html#cb8-29" tabindex="-1"></a><span class="do">## KEGG</span></span>
<span id="cb8-30"><a href="web-server-manual.html#cb8-30" tabindex="-1"></a><span class="co">#choose the metabolism pathway</span></span></code></pre></div>
<blockquote>
<strong>Extract the data of metabolism</strong>
</blockquote>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="web-server-manual.html#cb9-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb9-2"><a href="web-server-manual.html#cb9-2" tabindex="-1"></a><span class="fu">library</span>(MNet)</span>
<span id="cb9-3"><a href="web-server-manual.html#cb9-3" tabindex="-1"></a></span>
<span id="cb9-4"><a href="web-server-manual.html#cb9-4" tabindex="-1"></a>kegg_data_metabolism <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/Graphite/gene-metabolite-kegg.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-5"><a href="web-server-manual.html#cb9-5" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-6"><a href="web-server-manual.html#cb9-6" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> <span class="fu">unique</span>(kegg_pathway<span class="sc">$</span>PATHWAY)) <span class="sc">%&gt;%</span></span>
<span id="cb9-7"><a href="web-server-manual.html#cb9-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">&quot;direction&quot;</span>,<span class="st">&quot;type&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-8"><a href="web-server-manual.html#cb9-8" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-9"><a href="web-server-manual.html#cb9-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(src_type <span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span> <span class="sc">|</span> dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-10"><a href="web-server-manual.html#cb9-10" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">src_new =</span> <span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,src,dest),</span>
<span id="cb9-11"><a href="web-server-manual.html#cb9-11" tabindex="-1"></a>                <span class="at">dest_new =</span> <span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,dest,src),</span>
<span id="cb9-12"><a href="web-server-manual.html#cb9-12" tabindex="-1"></a>                <span class="at">src_type_new=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,src_type,dest_type),</span>
<span id="cb9-13"><a href="web-server-manual.html#cb9-13" tabindex="-1"></a>                <span class="at">dest_type_new=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,dest_type,src_type)) <span class="sc">%&gt;%</span></span>
<span id="cb9-14"><a href="web-server-manual.html#cb9-14" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">src_new1=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span> <span class="sc">&amp;</span> dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,</span>
<span id="cb9-15"><a href="web-server-manual.html#cb9-15" tabindex="-1"></a>                                <span class="fu">ifelse</span>(src_new<span class="sc">&gt;</span>dest_new,src_new,dest_new),</span>
<span id="cb9-16"><a href="web-server-manual.html#cb9-16" tabindex="-1"></a>                                src_new)) <span class="sc">%&gt;%</span></span>
<span id="cb9-17"><a href="web-server-manual.html#cb9-17" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">dest_new1=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span> <span class="sc">&amp;</span> dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,</span>
<span id="cb9-18"><a href="web-server-manual.html#cb9-18" tabindex="-1"></a>                                 <span class="fu">ifelse</span>(src_new<span class="sc">&gt;</span>dest_new,dest_new,src_new),</span>
<span id="cb9-19"><a href="web-server-manual.html#cb9-19" tabindex="-1"></a>                                 dest_new)) <span class="sc">%&gt;%</span></span>
<span id="cb9-20"><a href="web-server-manual.html#cb9-20" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">&quot;src_type&quot;</span>,<span class="st">&quot;src&quot;</span>,<span class="st">&quot;dest_type&quot;</span>,<span class="st">&quot;dest&quot;</span>,<span class="st">&quot;src_new&quot;</span>,<span class="st">&quot;dest_new&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-21"><a href="web-server-manual.html#cb9-21" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;src_type&quot;</span><span class="ot">=</span><span class="st">&quot;src_type_new&quot;</span>,<span class="st">&quot;src&quot;</span><span class="ot">=</span><span class="st">&quot;src_new1&quot;</span>,<span class="st">&quot;dest_type&quot;</span><span class="ot">=</span><span class="st">&quot;dest_type_new&quot;</span>,<span class="st">&quot;dest&quot;</span><span class="ot">=</span><span class="st">&quot;dest_new1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-22"><a href="web-server-manual.html#cb9-22" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&quot;src_type&quot;</span>,<span class="st">&quot;src&quot;</span>,<span class="st">&quot;dest_type&quot;</span>,<span class="st">&quot;dest&quot;</span>,<span class="st">&quot;pathwayid&quot;</span>,<span class="st">&quot;pathway&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-23"><a href="web-server-manual.html#cb9-23" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb9-24"><a href="web-server-manual.html#cb9-24" tabindex="-1"></a><span class="fu">write.table</span>(kegg_data_metabolism,<span class="st">&quot;result/Graphite/gene-metabolite-kegg_metabolism.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span>F)</span>
<span id="cb9-25"><a href="web-server-manual.html#cb9-25" tabindex="-1"></a></span>
<span id="cb9-26"><a href="web-server-manual.html#cb9-26" tabindex="-1"></a><span class="do">## SMPDB</span></span>
<span id="cb9-27"><a href="web-server-manual.html#cb9-27" tabindex="-1"></a>metabolism_pathway <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;result/Graphite/smpdb_pathways.csv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-28"><a href="web-server-manual.html#cb9-28" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Subject<span class="sc">==</span><span class="st">&quot;Metabolic&quot;</span>)</span>
<span id="cb9-29"><a href="web-server-manual.html#cb9-29" tabindex="-1"></a></span>
<span id="cb9-30"><a href="web-server-manual.html#cb9-30" tabindex="-1"></a>smpdb_data <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/Graphite/gene-metabolite-smpdb.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-31"><a href="web-server-manual.html#cb9-31" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb9-32"><a href="web-server-manual.html#cb9-32" tabindex="-1"></a>smpdb_metabolism <span class="ot">&lt;-</span>  smpdb_data <span class="sc">%&gt;%</span></span>
<span id="cb9-33"><a href="web-server-manual.html#cb9-33" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathway <span class="sc">%in%</span> metabolism_pathway<span class="sc">$</span>Name) <span class="sc">%&gt;%</span></span>
<span id="cb9-34"><a href="web-server-manual.html#cb9-34" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span><span class="sc">|</span>dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-35"><a href="web-server-manual.html#cb9-35" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;De Novo&quot;</span>,pathway)) <span class="sc">%&gt;%</span></span>
<span id="cb9-36"><a href="web-server-manual.html#cb9-36" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway_new=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;Phosphatidylcholine Biosynthesis&quot;</span>,pathway),</span>
<span id="cb9-37"><a href="web-server-manual.html#cb9-37" tabindex="-1"></a>                                   <span class="st">&quot;Phosphatidylcholine Biosynthesis&quot;</span>,pathway)) <span class="sc">%&gt;%</span></span>
<span id="cb9-38"><a href="web-server-manual.html#cb9-38" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway_new=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;Cardiolipin Biosynthesis&quot;</span>,pathway_new),</span>
<span id="cb9-39"><a href="web-server-manual.html#cb9-39" tabindex="-1"></a>                                   <span class="st">&quot;Cardiolipin Biosynthesis&quot;</span>,pathway_new)) <span class="sc">%&gt;%</span></span>
<span id="cb9-40"><a href="web-server-manual.html#cb9-40" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway_new=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;Phosphatidylethanolamine Biosynthesis&quot;</span>,pathway_new),</span>
<span id="cb9-41"><a href="web-server-manual.html#cb9-41" tabindex="-1"></a>                                   <span class="st">&quot;Phosphatidylethanolamine Biosynthesis&quot;</span>,pathway_new)) <span class="sc">%&gt;%</span></span>
<span id="cb9-42"><a href="web-server-manual.html#cb9-42" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">pathway_new=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;Mitochondrial Beta-Oxidation&quot;</span>,pathway_new),</span>
<span id="cb9-43"><a href="web-server-manual.html#cb9-43" tabindex="-1"></a>                                   <span class="st">&quot;Mitochondrial Beta-Oxidation&quot;</span>,pathway_new)) <span class="sc">%&gt;%</span></span>
<span id="cb9-44"><a href="web-server-manual.html#cb9-44" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="st">&quot;pathway&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-45"><a href="web-server-manual.html#cb9-45" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;pathway&quot;</span><span class="ot">=</span><span class="st">&quot;pathway_new&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-46"><a href="web-server-manual.html#cb9-46" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">src_new =</span> <span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,src,dest),</span>
<span id="cb9-47"><a href="web-server-manual.html#cb9-47" tabindex="-1"></a>                <span class="at">dest_new =</span> <span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,dest,src),</span>
<span id="cb9-48"><a href="web-server-manual.html#cb9-48" tabindex="-1"></a>                <span class="at">src_type_new=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,src_type,dest_type),</span>
<span id="cb9-49"><a href="web-server-manual.html#cb9-49" tabindex="-1"></a>                <span class="at">dest_type_new=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,dest_type,src_type)) <span class="sc">%&gt;%</span></span>
<span id="cb9-50"><a href="web-server-manual.html#cb9-50" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">src_new1=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span> <span class="sc">&amp;</span> dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,</span>
<span id="cb9-51"><a href="web-server-manual.html#cb9-51" tabindex="-1"></a>                                <span class="fu">ifelse</span>(src_new<span class="sc">&gt;</span>dest_new,src_new,dest_new),</span>
<span id="cb9-52"><a href="web-server-manual.html#cb9-52" tabindex="-1"></a>                                src_new)) <span class="sc">%&gt;%</span></span>
<span id="cb9-53"><a href="web-server-manual.html#cb9-53" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">dest_new1=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span> <span class="sc">&amp;</span> dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,</span>
<span id="cb9-54"><a href="web-server-manual.html#cb9-54" tabindex="-1"></a>                                 <span class="fu">ifelse</span>(src_new<span class="sc">&gt;</span>dest_new,dest_new,src_new),</span>
<span id="cb9-55"><a href="web-server-manual.html#cb9-55" tabindex="-1"></a>                                 dest_new)) <span class="sc">%&gt;%</span></span>
<span id="cb9-56"><a href="web-server-manual.html#cb9-56" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">&quot;src_type&quot;</span>,<span class="st">&quot;src&quot;</span>,<span class="st">&quot;dest_type&quot;</span>,<span class="st">&quot;dest&quot;</span>,<span class="st">&quot;src_new&quot;</span>,<span class="st">&quot;dest_new&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-57"><a href="web-server-manual.html#cb9-57" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;src_type&quot;</span><span class="ot">=</span><span class="st">&quot;src_type_new&quot;</span>,<span class="st">&quot;src&quot;</span><span class="ot">=</span><span class="st">&quot;src_new1&quot;</span>,<span class="st">&quot;dest_type&quot;</span><span class="ot">=</span><span class="st">&quot;dest_type_new&quot;</span>,<span class="st">&quot;dest&quot;</span><span class="ot">=</span><span class="st">&quot;dest_new1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-58"><a href="web-server-manual.html#cb9-58" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&quot;src_type&quot;</span>,<span class="st">&quot;src&quot;</span>,<span class="st">&quot;dest_type&quot;</span>,<span class="st">&quot;dest&quot;</span>,<span class="st">&quot;pathwayid&quot;</span>,<span class="st">&quot;pathway&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-59"><a href="web-server-manual.html#cb9-59" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb9-60"><a href="web-server-manual.html#cb9-60" tabindex="-1"></a><span class="fu">write.table</span>(smpdb_metabolism,<span class="st">&quot;result/Graphite/gene-metabolite-smpdb_metabolism.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span>F)</span>
<span id="cb9-61"><a href="web-server-manual.html#cb9-61" tabindex="-1"></a></span>
<span id="cb9-62"><a href="web-server-manual.html#cb9-62" tabindex="-1"></a><span class="do">## Reactome</span></span>
<span id="cb9-63"><a href="web-server-manual.html#cb9-63" tabindex="-1"></a><span class="co"># metabolism pathway</span></span>
<span id="cb9-64"><a href="web-server-manual.html#cb9-64" tabindex="-1"></a>reactome_metabolism_pathwayid <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/Graphite/ReactomePathwaysRelation.txt&quot;</span>,<span class="at">header=</span>F) <span class="sc">%&gt;%</span></span>
<span id="cb9-65"><a href="web-server-manual.html#cb9-65" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-66"><a href="web-server-manual.html#cb9-66" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(V1<span class="sc">==</span><span class="st">&quot;R-HSA-1430728&quot;</span>)</span>
<span id="cb9-67"><a href="web-server-manual.html#cb9-67" tabindex="-1"></a></span>
<span id="cb9-68"><a href="web-server-manual.html#cb9-68" tabindex="-1"></a>reactome_metabolism_data <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/Graphite/gene-metabolite-reactome.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-69"><a href="web-server-manual.html#cb9-69" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-70"><a href="web-server-manual.html#cb9-70" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathwayid <span class="sc">%in%</span> reactome_metabolism_pathwayid<span class="sc">$</span>V2) <span class="sc">%&gt;%</span></span>
<span id="cb9-71"><a href="web-server-manual.html#cb9-71" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span><span class="sc">|</span>dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-72"><a href="web-server-manual.html#cb9-72" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">src_new =</span> <span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,src,dest),</span>
<span id="cb9-73"><a href="web-server-manual.html#cb9-73" tabindex="-1"></a>                <span class="at">dest_new =</span> <span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,dest,src),</span>
<span id="cb9-74"><a href="web-server-manual.html#cb9-74" tabindex="-1"></a>                <span class="at">src_type_new=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,src_type,dest_type),</span>
<span id="cb9-75"><a href="web-server-manual.html#cb9-75" tabindex="-1"></a>                <span class="at">dest_type_new=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,dest_type,src_type)) <span class="sc">%&gt;%</span></span>
<span id="cb9-76"><a href="web-server-manual.html#cb9-76" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">src_new1=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span> <span class="sc">&amp;</span> dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,</span>
<span id="cb9-77"><a href="web-server-manual.html#cb9-77" tabindex="-1"></a>                                <span class="fu">ifelse</span>(src_new<span class="sc">&gt;</span>dest_new,src_new,dest_new),</span>
<span id="cb9-78"><a href="web-server-manual.html#cb9-78" tabindex="-1"></a>                                src_new)) <span class="sc">%&gt;%</span></span>
<span id="cb9-79"><a href="web-server-manual.html#cb9-79" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">dest_new1=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span> <span class="sc">&amp;</span> dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,</span>
<span id="cb9-80"><a href="web-server-manual.html#cb9-80" tabindex="-1"></a>                                 <span class="fu">ifelse</span>(src_new<span class="sc">&gt;</span>dest_new,dest_new,src_new),</span>
<span id="cb9-81"><a href="web-server-manual.html#cb9-81" tabindex="-1"></a>                                 dest_new)) <span class="sc">%&gt;%</span></span>
<span id="cb9-82"><a href="web-server-manual.html#cb9-82" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">&quot;src_type&quot;</span>,<span class="st">&quot;src&quot;</span>,<span class="st">&quot;dest_type&quot;</span>,<span class="st">&quot;dest&quot;</span>,<span class="st">&quot;src_new&quot;</span>,<span class="st">&quot;dest_new&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-83"><a href="web-server-manual.html#cb9-83" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;src_type&quot;</span><span class="ot">=</span><span class="st">&quot;src_type_new&quot;</span>,<span class="st">&quot;src&quot;</span><span class="ot">=</span><span class="st">&quot;src_new1&quot;</span>,<span class="st">&quot;dest_type&quot;</span><span class="ot">=</span><span class="st">&quot;dest_type_new&quot;</span>,<span class="st">&quot;dest&quot;</span><span class="ot">=</span><span class="st">&quot;dest_new1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-84"><a href="web-server-manual.html#cb9-84" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&quot;src_type&quot;</span>,<span class="st">&quot;src&quot;</span>,<span class="st">&quot;dest_type&quot;</span>,<span class="st">&quot;dest&quot;</span>,<span class="st">&quot;pathwayid&quot;</span>,<span class="st">&quot;pathway&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-85"><a href="web-server-manual.html#cb9-85" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb9-86"><a href="web-server-manual.html#cb9-86" tabindex="-1"></a><span class="fu">write.table</span>(reactome_metabolism_data,<span class="st">&quot;result/Graphite/gene-metabolite-reactome_metabolism.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span>F)</span>
<span id="cb9-87"><a href="web-server-manual.html#cb9-87" tabindex="-1"></a></span>
<span id="cb9-88"><a href="web-server-manual.html#cb9-88" tabindex="-1"></a><span class="do">## Wikipathways</span></span>
<span id="cb9-89"><a href="web-server-manual.html#cb9-89" tabindex="-1"></a>wikipathway_metabolism_id <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/wikipath/test_metabolic.txt&quot;</span>,<span class="at">header=</span>F) <span class="sc">%&gt;%</span></span>
<span id="cb9-90"><a href="web-server-manual.html#cb9-90" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb9-91"><a href="web-server-manual.html#cb9-91" tabindex="-1"></a></span>
<span id="cb9-92"><a href="web-server-manual.html#cb9-92" tabindex="-1"></a>wikipathway_data <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;result/Graphite/gene-metabolite-wikipaths.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-93"><a href="web-server-manual.html#cb9-93" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-94"><a href="web-server-manual.html#cb9-94" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathwayid <span class="sc">%in%</span> wikipathway_metabolism_id<span class="sc">$</span>V1) <span class="sc">%&gt;%</span></span>
<span id="cb9-95"><a href="web-server-manual.html#cb9-95" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span><span class="sc">|</span>dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-96"><a href="web-server-manual.html#cb9-96" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">src_new =</span> <span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,src,dest),</span>
<span id="cb9-97"><a href="web-server-manual.html#cb9-97" tabindex="-1"></a>                <span class="at">dest_new =</span> <span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,dest,src),</span>
<span id="cb9-98"><a href="web-server-manual.html#cb9-98" tabindex="-1"></a>                <span class="at">src_type_new=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,src_type,dest_type),</span>
<span id="cb9-99"><a href="web-server-manual.html#cb9-99" tabindex="-1"></a>                <span class="at">dest_type_new=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,dest_type,src_type)) <span class="sc">%&gt;%</span></span>
<span id="cb9-100"><a href="web-server-manual.html#cb9-100" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">src_new1=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span> <span class="sc">&amp;</span> dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,</span>
<span id="cb9-101"><a href="web-server-manual.html#cb9-101" tabindex="-1"></a>                                <span class="fu">ifelse</span>(src_new<span class="sc">&gt;</span>dest_new,src_new,dest_new),</span>
<span id="cb9-102"><a href="web-server-manual.html#cb9-102" tabindex="-1"></a>                                src_new)) <span class="sc">%&gt;%</span></span>
<span id="cb9-103"><a href="web-server-manual.html#cb9-103" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">dest_new1=</span><span class="fu">ifelse</span>(src_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span> <span class="sc">&amp;</span> dest_type<span class="sc">==</span><span class="st">&quot;KEGGCOMP&quot;</span>,</span>
<span id="cb9-104"><a href="web-server-manual.html#cb9-104" tabindex="-1"></a>                                 <span class="fu">ifelse</span>(src_new<span class="sc">&gt;</span>dest_new,dest_new,src_new),</span>
<span id="cb9-105"><a href="web-server-manual.html#cb9-105" tabindex="-1"></a>                                 dest_new)) <span class="sc">%&gt;%</span></span>
<span id="cb9-106"><a href="web-server-manual.html#cb9-106" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">&quot;src_type&quot;</span>,<span class="st">&quot;src&quot;</span>,<span class="st">&quot;dest_type&quot;</span>,<span class="st">&quot;dest&quot;</span>,<span class="st">&quot;src_new&quot;</span>,<span class="st">&quot;dest_new&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-107"><a href="web-server-manual.html#cb9-107" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;src_type&quot;</span><span class="ot">=</span><span class="st">&quot;src_type_new&quot;</span>,<span class="st">&quot;src&quot;</span><span class="ot">=</span><span class="st">&quot;src_new1&quot;</span>,<span class="st">&quot;dest_type&quot;</span><span class="ot">=</span><span class="st">&quot;dest_type_new&quot;</span>,<span class="st">&quot;dest&quot;</span><span class="ot">=</span><span class="st">&quot;dest_new1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-108"><a href="web-server-manual.html#cb9-108" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&quot;src_type&quot;</span>,<span class="st">&quot;src&quot;</span>,<span class="st">&quot;dest_type&quot;</span>,<span class="st">&quot;dest&quot;</span>,<span class="st">&quot;pathwayid&quot;</span>,<span class="st">&quot;pathway&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-109"><a href="web-server-manual.html#cb9-109" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb9-110"><a href="web-server-manual.html#cb9-110" tabindex="-1"></a><span class="fu">write.table</span>(wikipathway_data,<span class="st">&quot;result/Graphite/gene-metabolite-wikipathway_metabolism.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span>F)</span></code></pre></div>
<blockquote>
<strong>Combine the data from database kegg, wikipathway, reactome, smpdb and then uniq the data</strong>
</blockquote>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="web-server-manual.html#cb10-1" tabindex="-1"></a>cat gene<span class="sc">-</span>metabolite<span class="sc">-</span><span class="er">*</span>metabolism.txt <span class="sc">|</span>cut <span class="sc">-</span>f1<span class="dv">-4</span>,<span class="dv">6</span><span class="sc">|</span>sort<span class="sc">|</span>uniq <span class="sc">&gt;</span>result.graphite.txt</span></code></pre></div>
</div>
<div id="bigg" class="section level4 hasAnchor" number="2.3.2.3">
<h4><span class="header-section-number">2.3.2.3</span> BiGG<a href="web-server-manual.html#bigg" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<blockquote>
<strong>Down all the models in the BiGG</strong>
</blockquote>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="web-server-manual.html#cb11-1" tabindex="-1"></a><span class="do">## Down all the models in the BiGG</span></span>
<span id="cb11-2"><a href="web-server-manual.html#cb11-2" tabindex="-1"></a>curl <span class="st">&#39;http://bigg.ucsd.edu/api/v2/models&#39;</span></span>
<span id="cb11-3"><a href="web-server-manual.html#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="web-server-manual.html#cb11-4" tabindex="-1"></a><span class="do">## Then, choice the model from Homo sapiens,then in reserve model is iAT_PLT_636, iAB_RBC_283, RECON1, Recon3D.</span></span>
<span id="cb11-5"><a href="web-server-manual.html#cb11-5" tabindex="-1"></a></span>
<span id="cb11-6"><a href="web-server-manual.html#cb11-6" tabindex="-1"></a><span class="do">## Get all the reactions names in the 4 human models.</span></span>
<span id="cb11-7"><a href="web-server-manual.html#cb11-7" tabindex="-1"></a><span class="dv">2</span> minutes</span>
<span id="cb11-8"><a href="web-server-manual.html#cb11-8" tabindex="-1"></a></span>
<span id="cb11-9"><a href="web-server-manual.html#cb11-9" tabindex="-1"></a>curl <span class="st">&#39;http://bigg.ucsd.edu/api/v2/models/iAT_PLT_636/reactions&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>iAT_PLT_636.reactions</span>
<span id="cb11-10"><a href="web-server-manual.html#cb11-10" tabindex="-1"></a>curl <span class="st">&#39;http://bigg.ucsd.edu/api/v2/models/iAB_RBC_283/reactions&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>iAB_RBC_283.reactions</span>
<span id="cb11-11"><a href="web-server-manual.html#cb11-11" tabindex="-1"></a>curl <span class="st">&#39;http://bigg.ucsd.edu/api/v2/models/RECON1/reactions&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>RECON1.reactions</span>
<span id="cb11-12"><a href="web-server-manual.html#cb11-12" tabindex="-1"></a>curl <span class="st">&#39;http://bigg.ucsd.edu/api/v2/models/Recon3D/reactions&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>Recon3D.reactions</span>
<span id="cb11-13"><a href="web-server-manual.html#cb11-13" tabindex="-1"></a></span>
<span id="cb11-14"><a href="web-server-manual.html#cb11-14" tabindex="-1"></a><span class="do">## Change the json reaction names to txt file.</span></span>
<span id="cb11-15"><a href="web-server-manual.html#cb11-15" tabindex="-1"></a>tt<span class="ot">&lt;-</span>jsonlite<span class="sc">::</span><span class="fu">stream_in</span>(<span class="fu">file</span>(<span class="st">&quot;result/BiGG/iAB_RBC_283.reactions&quot;</span>),<span class="at">pagesize =</span> <span class="dv">100</span>)</span>
<span id="cb11-16"><a href="web-server-manual.html#cb11-16" tabindex="-1"></a><span class="fu">write.table</span>(tt<span class="sc">$</span>results[[<span class="dv">1</span>]]<span class="sc">$</span>bigg_id,<span class="st">&quot;result/BiGG/iAB_RBC_283.reaction.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">col.names=</span>F)</span>
<span id="cb11-17"><a href="web-server-manual.html#cb11-17" tabindex="-1"></a></span>
<span id="cb11-18"><a href="web-server-manual.html#cb11-18" tabindex="-1"></a>tt<span class="ot">&lt;-</span>jsonlite<span class="sc">::</span><span class="fu">stream_in</span>(<span class="fu">file</span>(<span class="st">&quot;result/BiGG/iAT_PLT_636.reactions&quot;</span>),<span class="at">pagesize =</span> <span class="dv">100</span>)</span>
<span id="cb11-19"><a href="web-server-manual.html#cb11-19" tabindex="-1"></a><span class="fu">write.table</span>(tt<span class="sc">$</span>results[[<span class="dv">1</span>]]<span class="sc">$</span>bigg_id,<span class="st">&quot;result/BiGG/iAT_PLT_636.reaction.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">col.names=</span>F)</span>
<span id="cb11-20"><a href="web-server-manual.html#cb11-20" tabindex="-1"></a></span>
<span id="cb11-21"><a href="web-server-manual.html#cb11-21" tabindex="-1"></a>tt<span class="ot">&lt;-</span>jsonlite<span class="sc">::</span><span class="fu">stream_in</span>(<span class="fu">file</span>(<span class="st">&quot;result/BiGG/RECON1.reactions&quot;</span>),<span class="at">pagesize =</span> <span class="dv">100</span>)</span>
<span id="cb11-22"><a href="web-server-manual.html#cb11-22" tabindex="-1"></a><span class="fu">write.table</span>(tt<span class="sc">$</span>results[[<span class="dv">1</span>]]<span class="sc">$</span>bigg_id,<span class="st">&quot;result/BiGG/RECON1.reaction.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">col.names=</span>F)</span>
<span id="cb11-23"><a href="web-server-manual.html#cb11-23" tabindex="-1"></a></span>
<span id="cb11-24"><a href="web-server-manual.html#cb11-24" tabindex="-1"></a>tt<span class="ot">&lt;-</span>jsonlite<span class="sc">::</span><span class="fu">stream_in</span>(<span class="fu">file</span>(<span class="st">&quot;result/BiGG/Recon3D.reactions&quot;</span>),<span class="at">pagesize =</span> <span class="dv">100</span>)</span>
<span id="cb11-25"><a href="web-server-manual.html#cb11-25" tabindex="-1"></a><span class="fu">write.table</span>(tt<span class="sc">$</span>results[[<span class="dv">1</span>]]<span class="sc">$</span>bigg_id,<span class="st">&quot;result/BiGG/Recon3D.reaction.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names=</span>F,<span class="at">col.names=</span>F)</span></code></pre></div>
<blockquote>
<strong>Download the every reaction</strong>
</blockquote>
<ul>
<li>
<p>Download the every reaction.</p>
<p>Download iAB_RBC_283 reaction</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="web-server-manual.html#cb12-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb12-2"><a href="web-server-manual.html#cb12-2" tabindex="-1"></a></span>
<span id="cb12-3"><a href="web-server-manual.html#cb12-3" tabindex="-1"></a><span class="co">#####Does output_dir need to add result/BiGG/mainly depends on the path of the full text</span></span>
<span id="cb12-4"><a href="web-server-manual.html#cb12-4" tabindex="-1"></a><span class="va">input_file</span><span class="op">=</span><span class="st">&#39;iAB_RBC_283.reaction.txt&#39;</span></span>
<span id="cb12-5"><a href="web-server-manual.html#cb12-5" tabindex="-1"></a><span class="va">output_dir</span><span class="op">=</span><span class="st">&#39;iAB_RBC_283/reaction/json&#39;</span></span>
<span id="cb12-6"><a href="web-server-manual.html#cb12-6" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">&quot;</span><span class="va">$output_dir</span><span class="st">&quot;</span></span>
<span id="cb12-7"><a href="web-server-manual.html#cb12-7" tabindex="-1"></a></span>
<span id="cb12-8"><a href="web-server-manual.html#cb12-8" tabindex="-1"></a><span class="fu">cat</span> <span class="st">&quot;</span><span class="va">$input_file</span><span class="st">&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span></span>
<span id="cb12-9"><a href="web-server-manual.html#cb12-9" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb12-10"><a href="web-server-manual.html#cb12-10" tabindex="-1"></a><span class="ex">curl</span> <span class="st">&quot;http://bigg.ucsd.edu/api/v2/models/iAB_RBC_283/reactions/</span><span class="va">${line}</span><span class="st">&quot;</span> <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${output_dir}</span><span class="st">/</span><span class="va">${line}</span><span class="st">.txt&quot;</span></span>
<span id="cb12-11"><a href="web-server-manual.html#cb12-11" tabindex="-1"></a><span class="cf">done</span> </span></code></pre></div>
<p>Download iAT_PLT_636 reaction</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="web-server-manual.html#cb13-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb13-2"><a href="web-server-manual.html#cb13-2" tabindex="-1"></a></span>
<span id="cb13-3"><a href="web-server-manual.html#cb13-3" tabindex="-1"></a><span class="co">#####Does output_dir need to add result/BiGG/mainly depends on the path of the full text</span></span>
<span id="cb13-4"><a href="web-server-manual.html#cb13-4" tabindex="-1"></a><span class="va">input_file</span><span class="op">=</span><span class="st">&#39;iAT_PLT_636.reaction.txt&#39;</span></span>
<span id="cb13-5"><a href="web-server-manual.html#cb13-5" tabindex="-1"></a><span class="va">output_dir</span><span class="op">=</span><span class="st">&#39;iAT_PLT_636/reaction/json&#39;</span></span>
<span id="cb13-6"><a href="web-server-manual.html#cb13-6" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">&quot;</span><span class="va">$output_dir</span><span class="st">&quot;</span></span>
<span id="cb13-7"><a href="web-server-manual.html#cb13-7" tabindex="-1"></a></span>
<span id="cb13-8"><a href="web-server-manual.html#cb13-8" tabindex="-1"></a><span class="fu">cat</span> <span class="st">&quot;</span><span class="va">$input_file</span><span class="st">&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span></span>
<span id="cb13-9"><a href="web-server-manual.html#cb13-9" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb13-10"><a href="web-server-manual.html#cb13-10" tabindex="-1"></a><span class="ex">curl</span> <span class="st">&quot;http://bigg.ucsd.edu/api/v2/models/iAT_PLT_636/reactions/</span><span class="va">${line}</span><span class="st">&quot;</span> <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${output_dir}</span><span class="st">/</span><span class="va">${line}</span><span class="st">.txt&quot;</span></span>
<span id="cb13-11"><a href="web-server-manual.html#cb13-11" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Download RECON1 reaction</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="web-server-manual.html#cb14-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb14-2"><a href="web-server-manual.html#cb14-2" tabindex="-1"></a></span>
<span id="cb14-3"><a href="web-server-manual.html#cb14-3" tabindex="-1"></a><span class="co">#####Does output_dir need to add result/BiGG/mainly depends on the path of the full text</span></span>
<span id="cb14-4"><a href="web-server-manual.html#cb14-4" tabindex="-1"></a><span class="va">input_file</span><span class="op">=</span><span class="st">&#39;RECON1.reaction.txt&#39;</span></span>
<span id="cb14-5"><a href="web-server-manual.html#cb14-5" tabindex="-1"></a><span class="va">output_dir</span><span class="op">=</span><span class="st">&#39;RECON1/reaction/json&#39;</span></span>
<span id="cb14-6"><a href="web-server-manual.html#cb14-6" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">&quot;</span><span class="va">$output_dir</span><span class="st">&quot;</span></span>
<span id="cb14-7"><a href="web-server-manual.html#cb14-7" tabindex="-1"></a></span>
<span id="cb14-8"><a href="web-server-manual.html#cb14-8" tabindex="-1"></a><span class="fu">cat</span> <span class="st">&quot;</span><span class="va">$input_file</span><span class="st">&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span></span>
<span id="cb14-9"><a href="web-server-manual.html#cb14-9" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb14-10"><a href="web-server-manual.html#cb14-10" tabindex="-1"></a><span class="ex">curl</span> <span class="st">&quot;http://bigg.ucsd.edu/api/v2/models/RECON1/reactions/</span><span class="va">${line}</span><span class="st">&quot;</span> <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${output_dir}</span><span class="st">/</span><span class="va">${line}</span><span class="st">.txt&quot;</span></span>
<span id="cb14-11"><a href="web-server-manual.html#cb14-11" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Download Recon3D reaction</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="web-server-manual.html#cb15-1" tabindex="-1"></a><span class="co">#!/bin/bash </span></span>
<span id="cb15-2"><a href="web-server-manual.html#cb15-2" tabindex="-1"></a></span>
<span id="cb15-3"><a href="web-server-manual.html#cb15-3" tabindex="-1"></a><span class="co">#####Does output_dir need to add result/BiGG/mainly depends on the path of the full text</span></span>
<span id="cb15-4"><a href="web-server-manual.html#cb15-4" tabindex="-1"></a><span class="va">input_file</span><span class="op">=</span><span class="st">&#39;Recon3D.reaction.txt&#39;</span></span>
<span id="cb15-5"><a href="web-server-manual.html#cb15-5" tabindex="-1"></a><span class="va">output_dir</span><span class="op">=</span><span class="st">&#39;Recon3D/reaction/json&#39;</span></span>
<span id="cb15-6"><a href="web-server-manual.html#cb15-6" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">&quot;</span><span class="va">$output_dir</span><span class="st">&quot;</span></span>
<span id="cb15-7"><a href="web-server-manual.html#cb15-7" tabindex="-1"></a></span>
<span id="cb15-8"><a href="web-server-manual.html#cb15-8" tabindex="-1"></a><span class="fu">cat</span> <span class="st">&quot;</span><span class="va">$input_file</span><span class="st">&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span>   </span>
<span id="cb15-9"><a href="web-server-manual.html#cb15-9" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb15-10"><a href="web-server-manual.html#cb15-10" tabindex="-1"></a><span class="ex">curl</span> <span class="st">&quot;http://bigg.ucsd.edu/api/v2/models/Recon3D/reactions/</span><span class="va">${line}</span><span class="st">&quot;</span> <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${output_dir}</span><span class="st">/</span><span class="va">${line}</span><span class="st">.txt&quot;</span></span>
<span id="cb15-11"><a href="web-server-manual.html#cb15-11" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<li>
<p>Get the information in every reaction.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="web-server-manual.html#cb16-1" tabindex="-1"></a><span class="do">## The R script that change the reaction information in json to txt.</span></span>
<span id="cb16-2"><a href="web-server-manual.html#cb16-2" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(T) </span>
<span id="cb16-3"><a href="web-server-manual.html#cb16-3" tabindex="-1"></a></span>
<span id="cb16-4"><a href="web-server-manual.html#cb16-4" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb16-5"><a href="web-server-manual.html#cb16-5" tabindex="-1"></a></span>
<span id="cb16-6"><a href="web-server-manual.html#cb16-6" tabindex="-1"></a>mydata <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;result/BiGG/&quot;</span>,args[<span class="dv">1</span>],<span class="st">&quot;/reaction/json/&quot;</span>,args[<span class="dv">2</span>],<span class="st">&quot;.txt&quot;</span>)</span>
<span id="cb16-7"><a href="web-server-manual.html#cb16-7" tabindex="-1"></a></span>
<span id="cb16-8"><a href="web-server-manual.html#cb16-8" tabindex="-1"></a>recon1<span class="ot">&lt;-</span>jsonlite<span class="sc">::</span><span class="fu">stream_in</span>(<span class="fu">file</span>(mydata),<span class="at">pagesize =</span> <span class="dv">100</span>)</span>
<span id="cb16-9"><a href="web-server-manual.html#cb16-9" tabindex="-1"></a></span>
<span id="cb16-10"><a href="web-server-manual.html#cb16-10" tabindex="-1"></a>metabolite_biggid <span class="ot">&lt;-</span> recon1<span class="sc">$</span>metabolites[[<span class="dv">1</span>]]<span class="sc">$</span>bigg_id</span>
<span id="cb16-11"><a href="web-server-manual.html#cb16-11" tabindex="-1"></a>metabolites <span class="ot">&lt;-</span> recon1<span class="sc">$</span>metabolites[[<span class="dv">1</span>]]<span class="sc">$</span>name</span>
<span id="cb16-12"><a href="web-server-manual.html#cb16-12" tabindex="-1"></a>compartment_bigg_id <span class="ot">&lt;-</span> recon1<span class="sc">$</span>metabolites[[<span class="dv">1</span>]]<span class="sc">$</span>compartment_bigg_id</span>
<span id="cb16-13"><a href="web-server-manual.html#cb16-13" tabindex="-1"></a></span>
<span id="cb16-14"><a href="web-server-manual.html#cb16-14" tabindex="-1"></a>gene <span class="ot">&lt;-</span> recon1<span class="sc">$</span>results[[<span class="dv">1</span>]]<span class="sc">$</span>genes[[<span class="dv">1</span>]]<span class="sc">$</span>name</span>
<span id="cb16-15"><a href="web-server-manual.html#cb16-15" tabindex="-1"></a></span>
<span id="cb16-16"><a href="web-server-manual.html#cb16-16" tabindex="-1"></a>subsystem <span class="ot">&lt;-</span> recon1<span class="sc">$</span>results[[<span class="dv">1</span>]]<span class="sc">$</span>subsystem</span>
<span id="cb16-17"><a href="web-server-manual.html#cb16-17" tabindex="-1"></a></span>
<span id="cb16-18"><a href="web-server-manual.html#cb16-18" tabindex="-1"></a>model<span class="ot">=</span>args[<span class="dv">1</span>]</span>
<span id="cb16-19"><a href="web-server-manual.html#cb16-19" tabindex="-1"></a></span>
<span id="cb16-20"><a href="web-server-manual.html#cb16-20" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(gene)<span class="sc">&gt;</span><span class="dv">0</span>) {</span>
<span id="cb16-21"><a href="web-server-manual.html#cb16-21" tabindex="-1"></a></span>
<span id="cb16-22"><a href="web-server-manual.html#cb16-22" tabindex="-1"></a>  dd <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">metabolite_biggid1=</span><span class="fu">paste</span>(metabolite_biggid,compartment_bigg_id,<span class="at">sep=</span><span class="st">&quot;_&quot;</span>),<span class="at">src_type=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">src=</span>metabolites,<span class="at">metabolite_biggid2=</span><span class="cn">NA</span>,<span class="at">dest_type=</span><span class="st">&quot;gene&quot;</span>,<span class="at">dest=</span><span class="fu">paste</span>(gene,<span class="at">collapse=</span><span class="st">&quot;;&quot;</span>),<span class="at">subsystems=</span>subsystem,<span class="at">models=</span>model)</span>
<span id="cb16-23"><a href="web-server-manual.html#cb16-23" tabindex="-1"></a></span>
<span id="cb16-24"><a href="web-server-manual.html#cb16-24" tabindex="-1"></a>}<span class="cf">else</span> {</span>
<span id="cb16-25"><a href="web-server-manual.html#cb16-25" tabindex="-1"></a>  dd <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb16-26"><a href="web-server-manual.html#cb16-26" tabindex="-1"></a>}</span>
<span id="cb16-27"><a href="web-server-manual.html#cb16-27" tabindex="-1"></a></span>
<span id="cb16-28"><a href="web-server-manual.html#cb16-28" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(<span class="fu">unique</span>(recon1<span class="sc">$</span>metabolites[[<span class="dv">1</span>]]<span class="sc">$</span>stoichiometry))<span class="sc">&gt;</span><span class="dv">1</span>) {</span>
<span id="cb16-29"><a href="web-server-manual.html#cb16-29" tabindex="-1"></a></span>
<span id="cb16-30"><a href="web-server-manual.html#cb16-30" tabindex="-1"></a>  metabolites_stoichiometry1 <span class="ot">&lt;-</span> recon1<span class="sc">$</span>metabolites[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb16-31"><a href="web-server-manual.html#cb16-31" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(stoichiometry<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb16-32"><a href="web-server-manual.html#cb16-32" tabindex="-1"></a></span>
<span id="cb16-33"><a href="web-server-manual.html#cb16-33" tabindex="-1"></a>  metabolites_stoichiometry2 <span class="ot">&lt;-</span> recon1<span class="sc">$</span>metabolites[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb16-34"><a href="web-server-manual.html#cb16-34" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(stoichiometry<span class="sc">&lt;</span><span class="dv">0</span>)</span>
<span id="cb16-35"><a href="web-server-manual.html#cb16-35" tabindex="-1"></a></span>
<span id="cb16-36"><a href="web-server-manual.html#cb16-36" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb16-37"><a href="web-server-manual.html#cb16-37" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(metabolites_stoichiometry1)) {</span>
<span id="cb16-38"><a href="web-server-manual.html#cb16-38" tabindex="-1"></a>    <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(metabolites_stoichiometry2)) {</span>
<span id="cb16-39"><a href="web-server-manual.html#cb16-39" tabindex="-1"></a>      temp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">metabolite_biggid1=</span><span class="fu">paste</span>(metabolites_stoichiometry1<span class="sc">$</span>bigg_id[i],metabolites_stoichiometry1<span class="sc">$</span>compartment_bigg_id[i],<span class="at">sep=</span><span class="st">&quot;_&quot;</span>),<span class="at">src_type=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">src=</span>metabolites_stoichiometry1<span class="sc">$</span>name[i],</span>
<span id="cb16-40"><a href="web-server-manual.html#cb16-40" tabindex="-1"></a>                         <span class="at">metabolite_biggid2=</span><span class="fu">paste</span>(metabolites_stoichiometry2<span class="sc">$</span>bigg_id[j],metabolites_stoichiometry2<span class="sc">$</span>compartment_bigg_id[j],<span class="at">sep=</span><span class="st">&quot;_&quot;</span>),<span class="at">dest_type=</span><span class="st">&quot;metabolite&quot;</span>,<span class="at">dest=</span>metabolites_stoichiometry2<span class="sc">$</span>name[j],<span class="at">subsystems=</span>subsystem,<span class="at">models=</span>model)</span>
<span id="cb16-41"><a href="web-server-manual.html#cb16-41" tabindex="-1"></a>      result <span class="ot">&lt;-</span> <span class="fu">rbind</span>(result,temp)</span>
<span id="cb16-42"><a href="web-server-manual.html#cb16-42" tabindex="-1"></a>    }</span>
<span id="cb16-43"><a href="web-server-manual.html#cb16-43" tabindex="-1"></a>  }</span>
<span id="cb16-44"><a href="web-server-manual.html#cb16-44" tabindex="-1"></a></span>
<span id="cb16-45"><a href="web-server-manual.html#cb16-45" tabindex="-1"></a>  result_final <span class="ot">&lt;-</span> <span class="fu">rbind</span>(dd,result)</span>
<span id="cb16-46"><a href="web-server-manual.html#cb16-46" tabindex="-1"></a></span>
<span id="cb16-47"><a href="web-server-manual.html#cb16-47" tabindex="-1"></a>  <span class="fu">write.table</span>(result_final,<span class="fu">paste0</span>(<span class="st">&quot;result/BiGG/&quot;</span>,args[<span class="dv">1</span>],<span class="st">&quot;/reaction/txt/&quot;</span>,args[<span class="dv">2</span>],<span class="st">&quot;.txt&quot;</span>),<span class="at">quote=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span>F)</span>
<span id="cb16-48"><a href="web-server-manual.html#cb16-48" tabindex="-1"></a>}<span class="cf">else</span> {</span>
<span id="cb16-49"><a href="web-server-manual.html#cb16-49" tabindex="-1"></a>  <span class="fu">print</span>(<span class="dv">0</span>)</span>
<span id="cb16-50"><a href="web-server-manual.html#cb16-50" tabindex="-1"></a>}</span></code></pre></div>
<li>
<p>Run the R script that change the reaction information in json to txt in batch.
This step takes about 3 hours.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="web-server-manual.html#cb17-1" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>iAB_RBC_283.reaction.txt <span class="sc">|</span><span class="cf">while</span> read line</span>
<span id="cb17-2"><a href="web-server-manual.html#cb17-2" tabindex="-1"></a>do</span>
<span id="cb17-3"><a href="web-server-manual.html#cb17-3" tabindex="-1"></a>Rscript reaction_json2txt.R iAB_RBC_283 <span class="sc">$</span>line</span>
<span id="cb17-4"><a href="web-server-manual.html#cb17-4" tabindex="-1"></a>done</span>
<span id="cb17-5"><a href="web-server-manual.html#cb17-5" tabindex="-1"></a></span>
<span id="cb17-6"><a href="web-server-manual.html#cb17-6" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>iAT_PLT_636.reaction.txt <span class="sc">|</span><span class="cf">while</span> read line</span>
<span id="cb17-7"><a href="web-server-manual.html#cb17-7" tabindex="-1"></a>do</span>
<span id="cb17-8"><a href="web-server-manual.html#cb17-8" tabindex="-1"></a>Rscript reaction_json2txt.R iAT_PLT_636 <span class="sc">$</span>line</span>
<span id="cb17-9"><a href="web-server-manual.html#cb17-9" tabindex="-1"></a>done</span>
<span id="cb17-10"><a href="web-server-manual.html#cb17-10" tabindex="-1"></a></span>
<span id="cb17-11"><a href="web-server-manual.html#cb17-11" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>RECON1.reaction.txt <span class="sc">|</span><span class="cf">while</span> read line</span>
<span id="cb17-12"><a href="web-server-manual.html#cb17-12" tabindex="-1"></a>do</span>
<span id="cb17-13"><a href="web-server-manual.html#cb17-13" tabindex="-1"></a>Rscript reaction_json2txt.R RECON1 <span class="sc">$</span>line</span>
<span id="cb17-14"><a href="web-server-manual.html#cb17-14" tabindex="-1"></a>done</span>
<span id="cb17-15"><a href="web-server-manual.html#cb17-15" tabindex="-1"></a></span>
<span id="cb17-16"><a href="web-server-manual.html#cb17-16" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>Recon3D.reaction.txt <span class="sc">|</span><span class="cf">while</span> read line</span>
<span id="cb17-17"><a href="web-server-manual.html#cb17-17" tabindex="-1"></a>do</span>
<span id="cb17-18"><a href="web-server-manual.html#cb17-18" tabindex="-1"></a>Rscript reaction_json2txt.R Recon3D <span class="sc">$</span>line</span>
<span id="cb17-19"><a href="web-server-manual.html#cb17-19" tabindex="-1"></a>done</span></code></pre></div>
<li>
<p>Combine all the reactions info include the gene-metabolite pair,the metabolite-metabolite pair,subsystem,model.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="web-server-manual.html#cb18-1" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>iAB_RBC_283<span class="sc">/</span>reaction<span class="sc">/</span>txt<span class="sc">/</span><span class="er">*|</span>grep <span class="sc">-</span>v metabolite_biggid1 <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>cat_reaction_info_iAB_RBC_283.txt</span>
<span id="cb18-2"><a href="web-server-manual.html#cb18-2" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>iAT_PLT_636<span class="sc">/</span>reaction<span class="sc">/</span>txt<span class="sc">/</span><span class="er">*|</span>grep <span class="sc">-</span>v metabolite_biggid1 <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>cat_reaction_info_iAT_PLT_636.txt</span>
<span id="cb18-3"><a href="web-server-manual.html#cb18-3" tabindex="-1"></a></span>
<span id="cb18-4"><a href="web-server-manual.html#cb18-4" tabindex="-1"></a><span class="do">## Recon3D manually modified several lines starting with (GlcNAc)7 (Man)3</span></span>
<span id="cb18-5"><a href="web-server-manual.html#cb18-5" tabindex="-1"></a></span>
<span id="cb18-6"><a href="web-server-manual.html#cb18-6" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>Recon3D<span class="sc">/</span>reaction<span class="sc">/</span>txt<span class="sc">/</span><span class="er">*|</span>grep <span class="sc">-</span>v metabolite_biggid1 <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>cat_reaction_info_Recon3D.txt</span>
<span id="cb18-7"><a href="web-server-manual.html#cb18-7" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>RECON1<span class="sc">/</span>reaction<span class="sc">/</span>txt<span class="sc">/</span><span class="er">*|</span>grep <span class="sc">-</span>v metabolite_biggid1 <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>cat_reaction_info_RECON1.txt</span></code></pre></div>
</ul>
<blockquote>
<strong>Exact all the metabolites names in every model in BiGG</strong>
</blockquote>
<ul>
<li>
<p>Dwonload the metabolites in every model in BiGG.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="web-server-manual.html#cb19-1" tabindex="-1"></a>curl <span class="st">&#39;http://bigg.ucsd.edu/api/v2/models/iAB_RBC_283/metabolites&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>iAB_RBC_283_metabolite.json</span>
<span id="cb19-2"><a href="web-server-manual.html#cb19-2" tabindex="-1"></a>curl <span class="st">&#39;http://bigg.ucsd.edu/api/v2/models/iAT_PLT_636/metabolites&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>iAT_PLT_636_metabolite.json</span>
<span id="cb19-3"><a href="web-server-manual.html#cb19-3" tabindex="-1"></a>curl <span class="st">&#39;http://bigg.ucsd.edu/api/v2/models/Recon3D/metabolites&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>Recon3D_metabolite.json</span>
<span id="cb19-4"><a href="web-server-manual.html#cb19-4" tabindex="-1"></a>curl <span class="st">&#39;http://bigg.ucsd.edu/api/v2/models/RECON1/metabolites&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>RECON1_metabolite.json</span></code></pre></div>
<li>
<p>Change the json to txt and extract metabolite bigg_id.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="web-server-manual.html#cb20-1" tabindex="-1"></a>args<span class="ot">=</span><span class="fu">commandArgs</span>(T)</span>
<span id="cb20-2"><a href="web-server-manual.html#cb20-2" tabindex="-1"></a>recon1<span class="ot">&lt;-</span>jsonlite<span class="sc">::</span><span class="fu">stream_in</span>(<span class="fu">file</span>(<span class="fu">paste0</span>(<span class="st">&quot;result/BiGG/&quot;</span>,args[<span class="dv">1</span>],<span class="st">&quot;_metabolite.json&quot;</span>)),<span class="at">pagesize =</span> <span class="dv">100</span>)</span>
<span id="cb20-3"><a href="web-server-manual.html#cb20-3" tabindex="-1"></a></span>
<span id="cb20-4"><a href="web-server-manual.html#cb20-4" tabindex="-1"></a>aa <span class="ot">&lt;-</span> recon1<span class="sc">$</span>results[[<span class="dv">1</span>]]</span>
<span id="cb20-5"><a href="web-server-manual.html#cb20-5" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">paste0</span>(aa<span class="sc">$</span>bigg_id,<span class="st">&quot;_&quot;</span>,aa<span class="sc">$</span>compartment_bigg_id),<span class="fu">paste0</span>(<span class="st">&quot;result/BiGG/&quot;</span>,args[<span class="dv">1</span>],<span class="st">&quot;_metabolite.txt&quot;</span>),<span class="at">row.names =</span> F,<span class="at">col.names=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">quote=</span>F)</span>
<span id="cb20-6"><a href="web-server-manual.html#cb20-6" tabindex="-1"></a></span>
<span id="cb20-7"><a href="web-server-manual.html#cb20-7" tabindex="-1"></a><span class="do">## Use the script.</span></span>
<span id="cb20-8"><a href="web-server-manual.html#cb20-8" tabindex="-1"></a>Rscript metabolitename_json2txt.R iAB_RBC_283</span>
<span id="cb20-9"><a href="web-server-manual.html#cb20-9" tabindex="-1"></a>Rscript metabolitename_json2txt.R iAT_PLT_636</span>
<span id="cb20-10"><a href="web-server-manual.html#cb20-10" tabindex="-1"></a>Rscript metabolitename_json2txt.R RECON1</span>
<span id="cb20-11"><a href="web-server-manual.html#cb20-11" tabindex="-1"></a>Rscript metabolitename_json2txt.R Recon3D</span></code></pre></div>
<li>
<p>Download every metabolite information.</p>
<p>Download iAB_RBC_283 metabolite</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="web-server-manual.html#cb21-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb21-2"><a href="web-server-manual.html#cb21-2" tabindex="-1"></a></span>
<span id="cb21-3"><a href="web-server-manual.html#cb21-3" tabindex="-1"></a><span class="co">#####Does output_dir need to add result/BiGG/mainly depends on the path of the full text</span></span>
<span id="cb21-4"><a href="web-server-manual.html#cb21-4" tabindex="-1"></a></span>
<span id="cb21-5"><a href="web-server-manual.html#cb21-5" tabindex="-1"></a><span class="va">input_file</span><span class="op">=</span><span class="st">&#39;iAB_RBC_283_metabolite.txt&#39;</span></span>
<span id="cb21-6"><a href="web-server-manual.html#cb21-6" tabindex="-1"></a><span class="va">output_dir</span><span class="op">=</span><span class="st">&#39;iAB_RBC_283/metabolite/json&#39;</span></span>
<span id="cb21-7"><a href="web-server-manual.html#cb21-7" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">&quot;</span><span class="va">$output_dir</span><span class="st">&quot;</span></span>
<span id="cb21-8"><a href="web-server-manual.html#cb21-8" tabindex="-1"></a></span>
<span id="cb21-9"><a href="web-server-manual.html#cb21-9" tabindex="-1"></a><span class="fu">cat</span> <span class="st">&quot;</span><span class="va">$input_file</span><span class="st">&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span></span>
<span id="cb21-10"><a href="web-server-manual.html#cb21-10" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb21-11"><a href="web-server-manual.html#cb21-11" tabindex="-1"></a><span class="ex">curl</span> <span class="st">&quot;http://bigg.ucsd.edu/api/v2/models/iAB_RBC_283/metabolites/</span><span class="va">${line}</span><span class="st">&quot;</span> <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${output_dir}</span><span class="st">/</span><span class="va">${line}</span><span class="st">.txt&quot;</span></span>
<span id="cb21-12"><a href="web-server-manual.html#cb21-12" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Download iAT_PLT_636 metabolite</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="web-server-manual.html#cb22-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb22-2"><a href="web-server-manual.html#cb22-2" tabindex="-1"></a></span>
<span id="cb22-3"><a href="web-server-manual.html#cb22-3" tabindex="-1"></a><span class="co">#####Does output_dir need to add result/BiGG/mainly depends on the path of the full text</span></span>
<span id="cb22-4"><a href="web-server-manual.html#cb22-4" tabindex="-1"></a></span>
<span id="cb22-5"><a href="web-server-manual.html#cb22-5" tabindex="-1"></a><span class="va">input_file</span><span class="op">=</span><span class="st">&#39;iAT_PLT_636_metabolite.txt&#39;</span></span>
<span id="cb22-6"><a href="web-server-manual.html#cb22-6" tabindex="-1"></a><span class="va">output_dir</span><span class="op">=</span><span class="st">&#39;iAT_PLT_636/metabolite/json&#39;</span></span>
<span id="cb22-7"><a href="web-server-manual.html#cb22-7" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">&quot;</span><span class="va">$output_dir</span><span class="st">&quot;</span></span>
<span id="cb22-8"><a href="web-server-manual.html#cb22-8" tabindex="-1"></a></span>
<span id="cb22-9"><a href="web-server-manual.html#cb22-9" tabindex="-1"></a><span class="fu">cat</span> <span class="st">&quot;</span><span class="va">$input_file</span><span class="st">&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span></span>
<span id="cb22-10"><a href="web-server-manual.html#cb22-10" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb22-11"><a href="web-server-manual.html#cb22-11" tabindex="-1"></a><span class="ex">curl</span> <span class="st">&quot;http://bigg.ucsd.edu/api/v2/models/iAT_PLT_636/metabolites/</span><span class="va">${line}</span><span class="st">&quot;</span> <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${output_dir}</span><span class="st">/</span><span class="va">${line}</span><span class="st">.txt&quot;</span></span>
<span id="cb22-12"><a href="web-server-manual.html#cb22-12" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Download RECON1 metabolite</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="web-server-manual.html#cb23-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb23-2"><a href="web-server-manual.html#cb23-2" tabindex="-1"></a></span>
<span id="cb23-3"><a href="web-server-manual.html#cb23-3" tabindex="-1"></a><span class="co">#####Does output_dir need to add result/BiGG/mainly depends on the path of the full text</span></span>
<span id="cb23-4"><a href="web-server-manual.html#cb23-4" tabindex="-1"></a></span>
<span id="cb23-5"><a href="web-server-manual.html#cb23-5" tabindex="-1"></a><span class="va">input_file</span><span class="op">=</span><span class="st">&#39;RECON1_metabolite.txt&#39;</span></span>
<span id="cb23-6"><a href="web-server-manual.html#cb23-6" tabindex="-1"></a><span class="va">output_dir</span><span class="op">=</span><span class="st">&#39;RECON1/metabolite/json&#39;</span></span>
<span id="cb23-7"><a href="web-server-manual.html#cb23-7" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">&quot;</span><span class="va">$output_dir</span><span class="st">&quot;</span></span>
<span id="cb23-8"><a href="web-server-manual.html#cb23-8" tabindex="-1"></a></span>
<span id="cb23-9"><a href="web-server-manual.html#cb23-9" tabindex="-1"></a><span class="fu">cat</span> <span class="st">&quot;</span><span class="va">$input_file</span><span class="st">&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span></span>
<span id="cb23-10"><a href="web-server-manual.html#cb23-10" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb23-11"><a href="web-server-manual.html#cb23-11" tabindex="-1"></a><span class="ex">curl</span> <span class="st">&quot;http://bigg.ucsd.edu/api/v2/models/RECON1/metabolites/</span><span class="va">${line}</span><span class="st">&quot;</span> <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${output_dir}</span><span class="st">/</span><span class="va">${line}</span><span class="st">.txt&quot;</span></span>
<span id="cb23-12"><a href="web-server-manual.html#cb23-12" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Download Recon3D metabolite</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="web-server-manual.html#cb24-1" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb24-2"><a href="web-server-manual.html#cb24-2" tabindex="-1"></a></span>
<span id="cb24-3"><a href="web-server-manual.html#cb24-3" tabindex="-1"></a><span class="co">#####Does output_dir need to add result/BiGG/mainly depends on the path of the full text</span></span>
<span id="cb24-4"><a href="web-server-manual.html#cb24-4" tabindex="-1"></a></span>
<span id="cb24-5"><a href="web-server-manual.html#cb24-5" tabindex="-1"></a><span class="va">input_file</span><span class="op">=</span><span class="st">&#39;Recon3D_metabolite.txt&#39;</span></span>
<span id="cb24-6"><a href="web-server-manual.html#cb24-6" tabindex="-1"></a><span class="va">output_dir</span><span class="op">=</span><span class="st">&#39;Recon3D/metabolite/json&#39;</span></span>
<span id="cb24-7"><a href="web-server-manual.html#cb24-7" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> <span class="st">&quot;</span><span class="va">$output_dir</span><span class="st">&quot;</span></span>
<span id="cb24-8"><a href="web-server-manual.html#cb24-8" tabindex="-1"></a></span>
<span id="cb24-9"><a href="web-server-manual.html#cb24-9" tabindex="-1"></a><span class="fu">cat</span> <span class="st">&quot;</span><span class="va">$input_file</span><span class="st">&quot;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">line</span></span>
<span id="cb24-10"><a href="web-server-manual.html#cb24-10" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb24-11"><a href="web-server-manual.html#cb24-11" tabindex="-1"></a><span class="ex">curl</span> <span class="st">&quot;http://bigg.ucsd.edu/api/v2/models/Recon3D/metabolites/</span><span class="va">${line}</span><span class="st">&quot;</span> <span class="op">&gt;</span> <span class="st">&quot;</span><span class="va">${output_dir}</span><span class="st">/</span><span class="va">${line}</span><span class="st">.txt&quot;</span></span>
<span id="cb24-12"><a href="web-server-manual.html#cb24-12" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<li>
<p>The R script that exact the KEGG ID in metabolite file.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="web-server-manual.html#cb25-1" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(T)</span>
<span id="cb25-2"><a href="web-server-manual.html#cb25-2" tabindex="-1"></a>mydata <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;result/BiGG/&quot;</span>,args[<span class="dv">1</span>],<span class="st">&quot;/metabolite/json/&quot;</span>,args[<span class="dv">2</span>],<span class="st">&quot;.json&quot;</span>)</span>
<span id="cb25-3"><a href="web-server-manual.html#cb25-3" tabindex="-1"></a><span class="co"># &quot;iAB_RBC_283_metabolites/glu__L_c.json&quot;</span></span>
<span id="cb25-4"><a href="web-server-manual.html#cb25-4" tabindex="-1"></a>recon1<span class="ot">&lt;-</span>jsonlite<span class="sc">::</span><span class="fu">stream_in</span>(<span class="fu">file</span>(mydata),<span class="at">pagesize =</span> <span class="dv">100</span>)</span>
<span id="cb25-5"><a href="web-server-manual.html#cb25-5" tabindex="-1"></a>kegg_id <span class="ot">&lt;-</span> <span class="fu">paste</span>(recon1<span class="sc">$</span>database_links<span class="sc">$</span><span class="st">`</span><span class="at">KEGG Compound</span><span class="st">`</span>[[<span class="dv">1</span>]]<span class="sc">$</span>id,<span class="at">collapse =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb25-6"><a href="web-server-manual.html#cb25-6" tabindex="-1"></a>name <span class="ot">&lt;-</span> recon1<span class="sc">$</span>name</span>
<span id="cb25-7"><a href="web-server-manual.html#cb25-7" tabindex="-1"></a></span>
<span id="cb25-8"><a href="web-server-manual.html#cb25-8" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">name=</span>name,<span class="at">kegg_id=</span>kegg_id,<span class="at">source=</span>args[<span class="dv">1</span>])</span>
<span id="cb25-9"><a href="web-server-manual.html#cb25-9" tabindex="-1"></a><span class="fu">write.table</span>(result,<span class="fu">paste0</span>(<span class="st">&quot;result/BiGG/&quot;</span>,args[<span class="dv">1</span>],<span class="st">&quot;/metabolite/txt/&quot;</span>,args[<span class="dv">2</span>],<span class="st">&quot;.txt&quot;</span>),<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<li>
<p>Run the R script that exact the KEGG ID in the metabolite file.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="web-server-manual.html#cb26-1" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>iAB_RBC_283_metabolite.txt<span class="sc">|</span><span class="cf">while</span> read line</span>
<span id="cb26-2"><a href="web-server-manual.html#cb26-2" tabindex="-1"></a>do</span>
<span id="cb26-3"><a href="web-server-manual.html#cb26-3" tabindex="-1"></a>  Rscript metabolite_json2txt.R iAB_RBC_283 <span class="sc">$</span>{line}</span>
<span id="cb26-4"><a href="web-server-manual.html#cb26-4" tabindex="-1"></a>done</span>
<span id="cb26-5"><a href="web-server-manual.html#cb26-5" tabindex="-1"></a></span>
<span id="cb26-6"><a href="web-server-manual.html#cb26-6" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>iAT_PLT_636_metabolite.txt <span class="sc">|</span><span class="cf">while</span> read line</span>
<span id="cb26-7"><a href="web-server-manual.html#cb26-7" tabindex="-1"></a>do</span>
<span id="cb26-8"><a href="web-server-manual.html#cb26-8" tabindex="-1"></a>  Rscript metabolite_json2txt.R iAT_PLT_636 <span class="sc">$</span>{line}</span>
<span id="cb26-9"><a href="web-server-manual.html#cb26-9" tabindex="-1"></a>done</span>
<span id="cb26-10"><a href="web-server-manual.html#cb26-10" tabindex="-1"></a></span>
<span id="cb26-11"><a href="web-server-manual.html#cb26-11" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>RECON1_metabolite.txt<span class="sc">|</span><span class="cf">while</span> read line</span>
<span id="cb26-12"><a href="web-server-manual.html#cb26-12" tabindex="-1"></a>do</span>
<span id="cb26-13"><a href="web-server-manual.html#cb26-13" tabindex="-1"></a>  Rscript metabolite_json2txt.R RECON1 <span class="sc">$</span>{line}</span>
<span id="cb26-14"><a href="web-server-manual.html#cb26-14" tabindex="-1"></a>done</span>
<span id="cb26-15"><a href="web-server-manual.html#cb26-15" tabindex="-1"></a></span>
<span id="cb26-16"><a href="web-server-manual.html#cb26-16" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>Recon3D_metabolite.txt<span class="sc">|</span><span class="cf">while</span> read line</span>
<span id="cb26-17"><a href="web-server-manual.html#cb26-17" tabindex="-1"></a>do</span>
<span id="cb26-18"><a href="web-server-manual.html#cb26-18" tabindex="-1"></a>  Rscript metabolite_json2txt.R Recon3D <span class="sc">$</span>{line}</span>
<span id="cb26-19"><a href="web-server-manual.html#cb26-19" tabindex="-1"></a>done</span></code></pre></div>
<li>
<p>Combine all the metabolite in 1 file for every model.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="web-server-manual.html#cb27-1" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>iAB_RBC_283<span class="sc">/</span>metabolite<span class="sc">/</span>txt<span class="sc">/</span><span class="er">*</span>.txt<span class="sc">|</span>grep <span class="sc">-</span>v kegg_id<span class="sc">|</span>grep <span class="sc">-</span>v <span class="st">&#39;^ $&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>iAB_RBC_283_metabolite_all.txt</span>
<span id="cb27-2"><a href="web-server-manual.html#cb27-2" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>iAT_PLT_636<span class="sc">/</span>metabolite<span class="sc">/</span>txt<span class="sc">/</span><span class="er">*</span>.txt <span class="sc">|</span>grep <span class="sc">-</span>v kegg_id <span class="sc">|</span>grep <span class="sc">-</span>v <span class="st">&#39;^ $&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>iAT_PLT_636_metabolite_all.txt</span>
<span id="cb27-3"><a href="web-server-manual.html#cb27-3" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>RECON1<span class="sc">/</span>metabolite<span class="sc">/</span>txt<span class="sc">/</span><span class="er">*</span>.txt<span class="sc">|</span>grep <span class="sc">-</span>v kegg_id<span class="sc">|</span>grep <span class="sc">-</span>v <span class="st">&#39;^ $&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>RECON1_metabolite_all.txt</span>
<span id="cb27-4"><a href="web-server-manual.html#cb27-4" tabindex="-1"></a></span>
<span id="cb27-5"><a href="web-server-manual.html#cb27-5" tabindex="-1"></a><span class="do">## Recon3D metabolites is so many, so please cat 2 times</span></span>
<span id="cb27-6"><a href="web-server-manual.html#cb27-6" tabindex="-1"></a>cat result<span class="sc">/</span>BiGG<span class="sc">/</span>Recon3D<span class="sc">/</span>metabolite<span class="sc">/</span>txt<span class="sc">/</span><span class="er">*</span>.txt<span class="sc">|</span>grep <span class="sc">-</span>v kegg_id<span class="sc">|</span>grep <span class="sc">-</span>v <span class="st">&#39;^ $&#39;</span> <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>Recon3D_metabolite_all.txt</span></code></pre></div>
</ul>
<blockquote>
<strong>The R script that change the metabolite name to kegg id in the final output</strong>
</blockquote>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="web-server-manual.html#cb28-1" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(T)</span>
<span id="cb28-2"><a href="web-server-manual.html#cb28-2" tabindex="-1"></a></span>
<span id="cb28-3"><a href="web-server-manual.html#cb28-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb28-4"><a href="web-server-manual.html#cb28-4" tabindex="-1"></a></span>
<span id="cb28-5"><a href="web-server-manual.html#cb28-5" tabindex="-1"></a>metabolite_info <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(<span class="st">&quot;result/BiGG/&quot;</span>,args[<span class="dv">1</span>],<span class="st">&quot;_metabolite_all.txt&quot;</span>),<span class="at">header=</span>F) <span class="sc">%&gt;%</span></span>
<span id="cb28-6"><a href="web-server-manual.html#cb28-6" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb28-7"><a href="web-server-manual.html#cb28-7" tabindex="-1"></a></span>
<span id="cb28-8"><a href="web-server-manual.html#cb28-8" tabindex="-1"></a>gene_metabolite_pairs <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(<span class="st">&quot;result/BiGG/cat_reaction_info_&quot;</span>,args[<span class="dv">1</span>],<span class="st">&quot;.txt&quot;</span>),<span class="at">header=</span>F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">fill=</span>T) <span class="sc">%&gt;%</span></span>
<span id="cb28-9"><a href="web-server-manual.html#cb28-9" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb28-10"><a href="web-server-manual.html#cb28-10" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(metabolite_info,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;V3&quot;</span><span class="ot">=</span><span class="st">&quot;V1&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb28-11"><a href="web-server-manual.html#cb28-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(metabolite_info,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;V6&quot;</span><span class="ot">=</span><span class="st">&quot;V1&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb28-12"><a href="web-server-manual.html#cb28-12" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">src=</span><span class="fu">ifelse</span>(V2.y<span class="sc">==</span><span class="st">&quot;&quot;</span> <span class="sc">|</span> <span class="fu">is.na</span>(V2.y),</span>
<span id="cb28-13"><a href="web-server-manual.html#cb28-13" tabindex="-1"></a>                           V3.x,V2.y)) <span class="sc">%&gt;%</span></span>
<span id="cb28-14"><a href="web-server-manual.html#cb28-14" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">dest=</span><span class="fu">ifelse</span>(V2<span class="sc">==</span><span class="st">&quot;&quot;</span> <span class="sc">|</span> <span class="fu">is.na</span>(V2),</span>
<span id="cb28-15"><a href="web-server-manual.html#cb28-15" tabindex="-1"></a>                            V6,V2)) <span class="sc">%&gt;%</span></span>
<span id="cb28-16"><a href="web-server-manual.html#cb28-16" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&quot;V2.x&quot;</span>,<span class="st">&quot;src&quot;</span>,<span class="st">&quot;V5&quot;</span>,<span class="st">&quot;dest&quot;</span>,<span class="st">&quot;V7&quot;</span>,<span class="st">&quot;V8&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb28-17"><a href="web-server-manual.html#cb28-17" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;src_type&quot;</span><span class="ot">=</span><span class="st">&quot;V2.x&quot;</span>,<span class="st">&quot;dest_type&quot;</span><span class="ot">=</span><span class="st">&quot;V5&quot;</span>,<span class="st">&quot;subsystems&quot;</span><span class="ot">=</span><span class="st">&quot;V7&quot;</span>,<span class="st">&quot;model&quot;</span><span class="ot">=</span><span class="st">&quot;V8&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-18"><a href="web-server-manual.html#cb28-18" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_rows</span>(src,<span class="at">sep=</span><span class="st">&quot;;&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-19"><a href="web-server-manual.html#cb28-19" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate_rows</span>(dest,<span class="at">sep=</span><span class="st">&quot;;&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-20"><a href="web-server-manual.html#cb28-20" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(src <span class="sc">!=</span> dest) <span class="sc">%&gt;%</span></span>
<span id="cb28-21"><a href="web-server-manual.html#cb28-21" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb28-22"><a href="web-server-manual.html#cb28-22" tabindex="-1"></a><span class="fu">write.table</span>(gene_metabolite_pairs,<span class="fu">paste0</span>(<span class="st">&quot;result/BiGG/result_&quot;</span>,args[<span class="dv">1</span>],<span class="st">&quot;.txt&quot;</span>),<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb28-23"><a href="web-server-manual.html#cb28-23" tabindex="-1"></a></span>
<span id="cb28-24"><a href="web-server-manual.html#cb28-24" tabindex="-1"></a><span class="do">## Use the R script that change the metabolite name to KEGG ID.</span></span>
<span id="cb28-25"><a href="web-server-manual.html#cb28-25" tabindex="-1"></a>Rscript result.R iAB_RBC_283</span>
<span id="cb28-26"><a href="web-server-manual.html#cb28-26" tabindex="-1"></a>Rscript result.R iAT_PLT_636</span>
<span id="cb28-27"><a href="web-server-manual.html#cb28-27" tabindex="-1"></a>Rscript result.R RECON1</span>
<span id="cb28-28"><a href="web-server-manual.html#cb28-28" tabindex="-1"></a>Rscript result.R Recon3D</span>
<span id="cb28-29"><a href="web-server-manual.html#cb28-29" tabindex="-1"></a>cat result_<span class="sc">*</span>.txt <span class="sc">|</span>cut <span class="sc">-</span>f1<span class="dv">-5</span> <span class="sc">|</span>sort<span class="sc">|</span>uniq <span class="sc">&gt;</span>result<span class="sc">/</span>BiGG<span class="sc">/</span>result.BiGG.txt</span></code></pre></div>
<blockquote>
**Combine the data from BiGG, graphite, and KEGG
</blockquote>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="web-server-manual.html#cb29-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb29-2"><a href="web-server-manual.html#cb29-2" tabindex="-1"></a></span>
<span id="cb29-3"><a href="web-server-manual.html#cb29-3" tabindex="-1"></a>kegg1 <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;KEGG_metabolite_gene.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-4"><a href="web-server-manual.html#cb29-4" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb29-5"><a href="web-server-manual.html#cb29-5" tabindex="-1"></a>kegg2 <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;KEGG_metabolite_metabolite.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-6"><a href="web-server-manual.html#cb29-6" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb29-7"><a href="web-server-manual.html#cb29-7" tabindex="-1"></a>dat_kegg <span class="ot">&lt;-</span> <span class="fu">rbind</span>(kegg1,kegg2)</span>
<span id="cb29-8"><a href="web-server-manual.html#cb29-8" tabindex="-1"></a></span>
<span id="cb29-9"><a href="web-server-manual.html#cb29-9" tabindex="-1"></a>dat_graphite <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;Graphite/result.graphite.txt&quot;</span>,<span class="at">header=</span>F) <span class="sc">%&gt;%</span></span>
<span id="cb29-10"><a href="web-server-manual.html#cb29-10" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb29-11"><a href="web-server-manual.html#cb29-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>V5) <span class="sc">%&gt;%</span></span>
<span id="cb29-12"><a href="web-server-manual.html#cb29-12" tabindex="-1"></a>  <span class="fu">filter</span>(V1 <span class="sc">!=</span> <span class="st">&quot;src_type&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-13"><a href="web-server-manual.html#cb29-13" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;src_type&quot;</span><span class="ot">=</span><span class="st">&quot;V1&quot;</span>,<span class="st">&quot;src&quot;</span><span class="ot">=</span><span class="st">&quot;V2&quot;</span>,<span class="st">&quot;dest_type&quot;</span><span class="ot">=</span><span class="st">&quot;V3&quot;</span>,<span class="st">&quot;dest&quot;</span><span class="ot">=</span><span class="st">&quot;V4&quot;</span>)</span>
<span id="cb29-14"><a href="web-server-manual.html#cb29-14" tabindex="-1"></a></span>
<span id="cb29-15"><a href="web-server-manual.html#cb29-15" tabindex="-1"></a>dat_bigg <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;BiGG/result.BiGG.txt&quot;</span>,<span class="at">header=</span>F) <span class="sc">%&gt;%</span></span>
<span id="cb29-16"><a href="web-server-manual.html#cb29-16" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb29-17"><a href="web-server-manual.html#cb29-17" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>V5) <span class="sc">%&gt;%</span></span>
<span id="cb29-18"><a href="web-server-manual.html#cb29-18" tabindex="-1"></a>  <span class="fu">filter</span>(V1 <span class="sc">!=</span> <span class="st">&quot;src_type&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-19"><a href="web-server-manual.html#cb29-19" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;src_type&quot;</span><span class="ot">=</span><span class="st">&quot;V1&quot;</span>,<span class="st">&quot;src&quot;</span><span class="ot">=</span><span class="st">&quot;V2&quot;</span>,<span class="st">&quot;dest_type&quot;</span><span class="ot">=</span><span class="st">&quot;V3&quot;</span>,<span class="st">&quot;dest&quot;</span><span class="ot">=</span><span class="st">&quot;V4&quot;</span>)</span>
<span id="cb29-20"><a href="web-server-manual.html#cb29-20" tabindex="-1"></a></span>
<span id="cb29-21"><a href="web-server-manual.html#cb29-21" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">rbind</span>(dat_kegg,dat_graphite,dat_bigg) </span>
<span id="cb29-22"><a href="web-server-manual.html#cb29-22" tabindex="-1"></a>dat1 <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span></span>
<span id="cb29-23"><a href="web-server-manual.html#cb29-23" tabindex="-1"></a>  <span class="fu">filter</span>(src_type<span class="sc">==</span>dest_type)</span>
<span id="cb29-24"><a href="web-server-manual.html#cb29-24" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span></span>
<span id="cb29-25"><a href="web-server-manual.html#cb29-25" tabindex="-1"></a>  <span class="fu">filter</span>(src_type<span class="sc">!=</span>dest_type)</span>
<span id="cb29-26"><a href="web-server-manual.html#cb29-26" tabindex="-1"></a></span>
<span id="cb29-27"><a href="web-server-manual.html#cb29-27" tabindex="-1"></a>dat11 <span class="ot">&lt;-</span> dat1 <span class="sc">%&gt;%</span></span>
<span id="cb29-28"><a href="web-server-manual.html#cb29-28" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">src1=</span><span class="fu">ifelse</span>( src <span class="sc">&gt;</span> dest,src,dest)) <span class="sc">%&gt;%</span></span>
<span id="cb29-29"><a href="web-server-manual.html#cb29-29" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">dest1=</span><span class="fu">ifelse</span>(src <span class="sc">&gt;</span> dest, dest, src)) <span class="sc">%&gt;%</span></span>
<span id="cb29-30"><a href="web-server-manual.html#cb29-30" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>src,<span class="sc">-</span>dest) <span class="sc">%&gt;%</span></span>
<span id="cb29-31"><a href="web-server-manual.html#cb29-31" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;src&quot;</span><span class="ot">=</span><span class="st">&quot;src1&quot;</span>,<span class="st">&quot;dest&quot;</span><span class="ot">=</span><span class="st">&quot;dest1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-32"><a href="web-server-manual.html#cb29-32" tabindex="-1"></a>  <span class="fu">select</span>(src_type,src,dest_type,dest)</span>
<span id="cb29-33"><a href="web-server-manual.html#cb29-33" tabindex="-1"></a>dat_all <span class="ot">&lt;-</span> <span class="fu">rbind</span>(dat2,dat11) <span class="sc">%&gt;%</span></span>
<span id="cb29-34"><a href="web-server-manual.html#cb29-34" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb29-35"><a href="web-server-manual.html#cb29-35" tabindex="-1"></a><span class="fu">write.table</span>(dat_all,<span class="st">&quot;Gene-Metabolite-data.txt&quot;</span>,<span class="at">quote=</span>F,<span class="at">row.names =</span> F,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
</div>
</div>
</div>
<div id="deployment" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Deployment<a href="web-server-manual.html#deployment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://github.com/tuantuangui/MNet">MNet Source Files</a></p>
<p><a href="http://www.mnet4all.com/MNet/">MNet Web Server</a></p>
<div id="installing-r-4.4.1-for-ubuntu-20" class="section level3 hasAnchor" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Installing R 4.4.1 for Ubuntu 20<a href="web-server-manual.html#installing-r-4.4.1-for-ubuntu-20" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="web-server-manual.html#cb30-1" tabindex="-1"></a><span class="co">## 1. Sources</span></span>
<span id="cb30-2"><a href="web-server-manual.html#cb30-2" tabindex="-1"></a><span class="fu">sudo</span> sh <span class="at">-c</span> <span class="st">&#39;echo &quot;deb https://mirrors.tuna.tsinghua.edu.cn/CRAN/bin/linux/ubuntu $(lsb_release -cs)-cran40/&quot; &gt;&gt; /etc/apt/sources.list&#39;</span></span>
<span id="cb30-3"><a href="web-server-manual.html#cb30-3" tabindex="-1"></a></span>
<span id="cb30-4"><a href="web-server-manual.html#cb30-4" tabindex="-1"></a><span class="fu">sudo</span> apt-key adv <span class="at">--keyserver</span> keyserver.ubuntu.com <span class="at">--recv-keys</span> E298A3A825C0D65DFD57CBB651716619E084DAB9</span>
<span id="cb30-5"><a href="web-server-manual.html#cb30-5" tabindex="-1"></a></span>
<span id="cb30-6"><a href="web-server-manual.html#cb30-6" tabindex="-1"></a><span class="co">## 2. Update and Install</span></span>
<span id="cb30-7"><a href="web-server-manual.html#cb30-7" tabindex="-1"></a><span class="fu">sudo</span> apt-get update</span>
<span id="cb30-8"><a href="web-server-manual.html#cb30-8" tabindex="-1"></a><span class="fu">sudo</span> apt install <span class="at">-y</span> libcurl4-openssl-dev libssl-dev libxml2-dev</span>
<span id="cb30-9"><a href="web-server-manual.html#cb30-9" tabindex="-1"></a><span class="fu">sudo</span> apt-get install r-base</span></code></pre></div>
</div>
<div id="installing-r-packages" class="section level3 hasAnchor" number="2.4.2">
<h3><span class="header-section-number">2.4.2</span> Installing R Packages<a href="web-server-manual.html#installing-r-packages" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb31-1"><a href="web-server-manual.html#cb31-1" tabindex="-1"></a><span class="co">## In Order Install</span></span>
<span id="cb31-2"><a href="web-server-manual.html#cb31-2" tabindex="-1"></a><span class="fu">sudo</span> R</span>
<span id="cb31-3"><a href="web-server-manual.html#cb31-3" tabindex="-1"></a></span>
<span id="cb31-4"><a href="web-server-manual.html#cb31-4" tabindex="-1"></a><span class="ex">options</span><span class="er">(</span><span class="ex">repos</span> = c<span class="er">(</span><span class="ex">CRAN</span> = <span class="st">&quot;https://mirrors.tuna.tsinghua.edu.cn/CRAN/&quot;</span><span class="kw">))</span></span>
<span id="cb31-5"><a href="web-server-manual.html#cb31-5" tabindex="-1"></a><span class="ex">options</span><span class="er">(</span><span class="va">BioC_mirror</span><span class="op">=</span><span class="st">&quot;https://mirrors.tuna.tsinghua.edu.cn/bioconductor&quot;</span><span class="kw">)</span></span>
<span id="cb31-6"><a href="web-server-manual.html#cb31-6" tabindex="-1"></a></span>
<span id="cb31-7"><a href="web-server-manual.html#cb31-7" tabindex="-1"></a><span class="ex">install.packages</span><span class="er">(</span><span class="st">&quot;shiny&quot;</span><span class="kw">)</span></span>
<span id="cb31-8"><a href="web-server-manual.html#cb31-8" tabindex="-1"></a><span class="ex">install.packages</span><span class="er">(</span><span class="st">&quot;Rcpp&quot;</span><span class="kw">)</span></span>
<span id="cb31-9"><a href="web-server-manual.html#cb31-9" tabindex="-1"></a><span class="ex">install.packages</span><span class="er">(</span><span class="st">&quot;RcppEigen&quot;</span><span class="kw">)</span></span>
<span id="cb31-10"><a href="web-server-manual.html#cb31-10" tabindex="-1"></a></span>
<span id="cb31-11"><a href="web-server-manual.html#cb31-11" tabindex="-1"></a><span class="ex">install.packages</span><span class="er">(</span><span class="st">&quot;BiocManager&quot;</span><span class="kw">)</span></span>
<span id="cb31-12"><a href="web-server-manual.html#cb31-12" tabindex="-1"></a><span class="ex">BiocManager::install</span><span class="er">(</span><span class="ex">c</span><span class="er">(</span><span class="st">&quot;supraHex&quot;</span><span class="ex">,</span> <span class="st">&quot;limma&quot;</span>, <span class="st">&quot;pathview&quot;</span>, <span class="st">&quot;ropls&quot;</span>, <span class="st">&quot;fgsea&quot;</span><span class="kw">))</span></span>
<span id="cb31-13"><a href="web-server-manual.html#cb31-13" tabindex="-1"></a></span>
<span id="cb31-14"><a href="web-server-manual.html#cb31-14" tabindex="-1"></a><span class="ex">install.packages</span><span class="er">(</span><span class="st">&quot;remotes&quot;</span><span class="kw">)</span> </span>
<span id="cb31-15"><a href="web-server-manual.html#cb31-15" tabindex="-1"></a><span class="ex">remotes::install_github</span><span class="er">(</span><span class="st">&quot;hfang-bristol/dnet&quot;</span><span class="kw">)</span></span>
<span id="cb31-16"><a href="web-server-manual.html#cb31-16" tabindex="-1"></a><span class="ex">remotes::install_github</span><span class="er">(</span><span class="st">&quot;hfang-bristol/XGR&quot;</span><span class="kw">)</span></span>
<span id="cb31-17"><a href="web-server-manual.html#cb31-17" tabindex="-1"></a></span>
<span id="cb31-18"><a href="web-server-manual.html#cb31-18" tabindex="-1"></a><span class="ex">remotes::install_github</span><span class="er">(</span><span class="st">&quot;tuantuangui/MNet&quot;</span><span class="kw">)</span></span></code></pre></div>
</div>
<div id="installing-shiny-sever" class="section level3 hasAnchor" number="2.4.3">
<h3><span class="header-section-number">2.4.3</span> Installing shiny-sever<a href="web-server-manual.html#installing-shiny-sever" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb32-1"><a href="web-server-manual.html#cb32-1" tabindex="-1"></a><span class="co">## 1. Install</span></span>
<span id="cb32-2"><a href="web-server-manual.html#cb32-2" tabindex="-1"></a><span class="fu">sudo</span> apt-get install gdebi-core</span>
<span id="cb32-3"><a href="web-server-manual.html#cb32-3" tabindex="-1"></a><span class="fu">wget</span> https://download3.rstudio.org/ubuntu-14.04/x86_64/shiny-server-1.5.14.948-amd64.deb</span>
<span id="cb32-4"><a href="web-server-manual.html#cb32-4" tabindex="-1"></a><span class="fu">sudo</span> gdebi shiny-server-1.5.14.948-amd64.deb</span>
<span id="cb32-5"><a href="web-server-manual.html#cb32-5" tabindex="-1"></a></span>
<span id="cb32-6"><a href="web-server-manual.html#cb32-6" tabindex="-1"></a><span class="co">## 2. Start and Enable shiny-server</span></span>
<span id="cb32-7"><a href="web-server-manual.html#cb32-7" tabindex="-1"></a><span class="fu">sudo</span> systemctl start shiny-server.service</span>
<span id="cb32-8"><a href="web-server-manual.html#cb32-8" tabindex="-1"></a><span class="fu">sudo</span> systemctl enable shiny-server</span>
<span id="cb32-9"><a href="web-server-manual.html#cb32-9" tabindex="-1"></a></span>
<span id="cb32-10"><a href="web-server-manual.html#cb32-10" tabindex="-1"></a><span class="co">## 3. Config</span></span>
<span id="cb32-11"><a href="web-server-manual.html#cb32-11" tabindex="-1"></a><span class="fu">sudo</span> vim /etc/shiny-server/shiny-server.conf</span>
<span id="cb32-12"><a href="web-server-manual.html#cb32-12" tabindex="-1"></a></span>
<span id="cb32-13"><a href="web-server-manual.html#cb32-13" tabindex="-1"></a><span class="co"># Instruct Shiny Server to run applications as the user &quot;shiny&quot;</span></span>
<span id="cb32-14"><a href="web-server-manual.html#cb32-14" tabindex="-1"></a><span class="ex">run_as</span> shiny<span class="kw">;</span></span>
<span id="cb32-15"><a href="web-server-manual.html#cb32-15" tabindex="-1"></a></span>
<span id="cb32-16"><a href="web-server-manual.html#cb32-16" tabindex="-1"></a><span class="co"># Define a server that listens on port 3838</span></span>
<span id="cb32-17"><a href="web-server-manual.html#cb32-17" tabindex="-1"></a><span class="ex">server</span> {</span>
<span id="cb32-18"><a href="web-server-manual.html#cb32-18" tabindex="-1"></a>  <span class="ex">listen</span> 3838<span class="kw">;</span></span>
<span id="cb32-19"><a href="web-server-manual.html#cb32-19" tabindex="-1"></a>  </span>
<span id="cb32-20"><a href="web-server-manual.html#cb32-20" tabindex="-1"></a><span class="co"># Define a location at the base URL</span></span>
<span id="cb32-21"><a href="web-server-manual.html#cb32-21" tabindex="-1"></a><span class="ex">location</span> / {</span>
<span id="cb32-22"><a href="web-server-manual.html#cb32-22" tabindex="-1"></a></span>
<span id="cb32-23"><a href="web-server-manual.html#cb32-23" tabindex="-1"></a><span class="co"># Host the directory of Shiny Apps stored in this directory</span></span>
<span id="cb32-24"><a href="web-server-manual.html#cb32-24" tabindex="-1"></a><span class="ex">site_dir</span> /srv/shiny-server<span class="kw">;</span></span>
<span id="cb32-25"><a href="web-server-manual.html#cb32-25" tabindex="-1"></a></span>
<span id="cb32-26"><a href="web-server-manual.html#cb32-26" tabindex="-1"></a><span class="co"># Log all Shiny output to files in this directory</span></span>
<span id="cb32-27"><a href="web-server-manual.html#cb32-27" tabindex="-1"></a><span class="ex">log_dir</span> /var/log/shiny-server<span class="kw">;</span></span>
<span id="cb32-28"><a href="web-server-manual.html#cb32-28" tabindex="-1"></a></span>
<span id="cb32-29"><a href="web-server-manual.html#cb32-29" tabindex="-1"></a><span class="co"># When a user visits the base URL rather than a particular application,</span></span>
<span id="cb32-30"><a href="web-server-manual.html#cb32-30" tabindex="-1"></a><span class="co"># an index of the applications available in this directory will be shown.</span></span>
<span id="cb32-31"><a href="web-server-manual.html#cb32-31" tabindex="-1"></a><span class="ex">directory_index</span> on<span class="kw">;</span></span>
<span id="cb32-32"><a href="web-server-manual.html#cb32-32" tabindex="-1"></a>  </span>
<span id="cb32-33"><a href="web-server-manual.html#cb32-33" tabindex="-1"></a><span class="ex">app_idle_timeout</span> 0<span class="kw">;</span></span>
<span id="cb32-34"><a href="web-server-manual.html#cb32-34" tabindex="-1"></a><span class="er">}</span></span>
<span id="cb32-35"><a href="web-server-manual.html#cb32-35" tabindex="-1"></a><span class="er">}</span></span></code></pre></div>
</div>
<div id="shiny-app-server-and-logs" class="section level3 hasAnchor" number="2.4.4">
<h3><span class="header-section-number">2.4.4</span> Shiny App Server and Logs<a href="web-server-manual.html#shiny-app-server-and-logs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb33"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb33-1"><a href="web-server-manual.html#cb33-1" tabindex="-1"></a><span class="co">## 1. Shinyapp</span></span>
<span id="cb33-2"><a href="web-server-manual.html#cb33-2" tabindex="-1"></a><span class="bu">cd</span> ~/</span>
<span id="cb33-3"><a href="web-server-manual.html#cb33-3" tabindex="-1"></a><span class="fu">git</span> clone https://github.com/tuantuangui/MNet.git</span>
<span id="cb33-4"><a href="web-server-manual.html#cb33-4" tabindex="-1"></a><span class="fu">sudo</span> cp <span class="at">-r</span> ~/MNet/inst/shinyapp/ /srv/shiny-server/MNet/</span>
<span id="cb33-5"><a href="web-server-manual.html#cb33-5" tabindex="-1"></a></span>
<span id="cb33-6"><a href="web-server-manual.html#cb33-6" tabindex="-1"></a><span class="co">## 2. Restart shiny-server</span></span>
<span id="cb33-7"><a href="web-server-manual.html#cb33-7" tabindex="-1"></a><span class="fu">sudo</span> systemctl restart shiny-server.service</span>
<span id="cb33-8"><a href="web-server-manual.html#cb33-8" tabindex="-1"></a></span>
<span id="cb33-9"><a href="web-server-manual.html#cb33-9" tabindex="-1"></a><span class="co">## 3. Less New Logs</span></span>
<span id="cb33-10"><a href="web-server-manual.html#cb33-10" tabindex="-1"></a><span class="fu">sudo</span> less /var/log/shiny-server/<span class="pp">*</span>.log</span></code></pre></div>
</div>
<div id="firewall-firewalld-or-ufw" class="section level3 hasAnchor" number="2.4.5">
<h3><span class="header-section-number">2.4.5</span> Firewall: firewalld or ufw<a href="web-server-manual.html#firewall-firewalld-or-ufw" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb34"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb34-1"><a href="web-server-manual.html#cb34-1" tabindex="-1"></a><span class="co">## firewalld</span></span>
<span id="cb34-2"><a href="web-server-manual.html#cb34-2" tabindex="-1"></a><span class="co"># 1. Install</span></span>
<span id="cb34-3"><a href="web-server-manual.html#cb34-3" tabindex="-1"></a><span class="fu">sudo</span> apt-get install firewalld</span>
<span id="cb34-4"><a href="web-server-manual.html#cb34-4" tabindex="-1"></a></span>
<span id="cb34-5"><a href="web-server-manual.html#cb34-5" tabindex="-1"></a><span class="co"># 2. Start and Enable </span></span>
<span id="cb34-6"><a href="web-server-manual.html#cb34-6" tabindex="-1"></a><span class="fu">sudo</span> systemctl start firewalld</span>
<span id="cb34-7"><a href="web-server-manual.html#cb34-7" tabindex="-1"></a><span class="fu">sudo</span> systemctl enable firewalld</span>
<span id="cb34-8"><a href="web-server-manual.html#cb34-8" tabindex="-1"></a></span>
<span id="cb34-9"><a href="web-server-manual.html#cb34-9" tabindex="-1"></a><span class="co"># 3. Usage</span></span>
<span id="cb34-10"><a href="web-server-manual.html#cb34-10" tabindex="-1"></a><span class="fu">sudo</span> firewall-cmd <span class="at">--list-ports</span></span>
<span id="cb34-11"><a href="web-server-manual.html#cb34-11" tabindex="-1"></a><span class="fu">sudo</span> firewall-cmd <span class="at">--zone</span><span class="op">=</span>public <span class="at">--add-port</span><span class="op">=</span>3838/tcp —-permanent</span>
<span id="cb34-12"><a href="web-server-manual.html#cb34-12" tabindex="-1"></a><span class="fu">sudo</span> firewall-cmd <span class="at">--reload</span></span>
<span id="cb34-13"><a href="web-server-manual.html#cb34-13" tabindex="-1"></a><span class="fu">sudo</span> firewall-cd <span class="at">--list-ports</span></span>
<span id="cb34-14"><a href="web-server-manual.html#cb34-14" tabindex="-1"></a></span>
<span id="cb34-15"><a href="web-server-manual.html#cb34-15" tabindex="-1"></a><span class="co">## ufw</span></span>
<span id="cb34-16"><a href="web-server-manual.html#cb34-16" tabindex="-1"></a><span class="co"># 1. Install</span></span>
<span id="cb34-17"><a href="web-server-manual.html#cb34-17" tabindex="-1"></a><span class="fu">sudo</span> apt-get install ufw</span>
<span id="cb34-18"><a href="web-server-manual.html#cb34-18" tabindex="-1"></a></span>
<span id="cb34-19"><a href="web-server-manual.html#cb34-19" tabindex="-1"></a><span class="co"># 2. Enable</span></span>
<span id="cb34-20"><a href="web-server-manual.html#cb34-20" tabindex="-1"></a><span class="fu">sudo</span> ufw enable</span>
<span id="cb34-21"><a href="web-server-manual.html#cb34-21" tabindex="-1"></a></span>
<span id="cb34-22"><a href="web-server-manual.html#cb34-22" tabindex="-1"></a><span class="co"># 3. Usage</span></span>
<span id="cb34-23"><a href="web-server-manual.html#cb34-23" tabindex="-1"></a><span class="fu">sudo</span> ufw status</span>
<span id="cb34-24"><a href="web-server-manual.html#cb34-24" tabindex="-1"></a><span class="fu">sudo</span> ufw allow 3838/tcp</span>
<span id="cb34-25"><a href="web-server-manual.html#cb34-25" tabindex="-1"></a><span class="fu">sudo</span> ufw reload</span>
<span id="cb34-26"><a href="web-server-manual.html#cb34-26" tabindex="-1"></a></span>
<span id="cb34-27"><a href="web-server-manual.html#cb34-27" tabindex="-1"></a><span class="fu">sudo</span> ufw delete allow 3838/tcp</span></code></pre></div>
</div>
<div id="apache2-config" class="section level3 hasAnchor" number="2.4.6">
<h3><span class="header-section-number">2.4.6</span> Apache2 Config<a href="web-server-manual.html#apache2-config" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb35"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb35-1"><a href="web-server-manual.html#cb35-1" tabindex="-1"></a><span class="co">## 1. Install </span></span>
<span id="cb35-2"><a href="web-server-manual.html#cb35-2" tabindex="-1"></a><span class="fu">sudo</span> apt-get install apache2 </span>
<span id="cb35-3"><a href="web-server-manual.html#cb35-3" tabindex="-1"></a><span class="fu">sudo</span> systemctl start apache2</span>
<span id="cb35-4"><a href="web-server-manual.html#cb35-4" tabindex="-1"></a><span class="fu">sudo</span> systemctl enable apache2</span>
<span id="cb35-5"><a href="web-server-manual.html#cb35-5" tabindex="-1"></a></span>
<span id="cb35-6"><a href="web-server-manual.html#cb35-6" tabindex="-1"></a><span class="co">## 2. a2enmod config </span></span>
<span id="cb35-7"><a href="web-server-manual.html#cb35-7" tabindex="-1"></a><span class="fu">sudo</span> a2enmod proxy</span>
<span id="cb35-8"><a href="web-server-manual.html#cb35-8" tabindex="-1"></a><span class="fu">sudo</span> a2enmod proxy_http </span>
<span id="cb35-9"><a href="web-server-manual.html#cb35-9" tabindex="-1"></a><span class="fu">sudo</span> a2enmod rewrite</span>
<span id="cb35-10"><a href="web-server-manual.html#cb35-10" tabindex="-1"></a><span class="fu">sudo</span> systemctl restart apache2</span>
<span id="cb35-11"><a href="web-server-manual.html#cb35-11" tabindex="-1"></a></span>
<span id="cb35-12"><a href="web-server-manual.html#cb35-12" tabindex="-1"></a><span class="co">## 3. Apache2 Config for shiny-server</span></span>
<span id="cb35-13"><a href="web-server-manual.html#cb35-13" tabindex="-1"></a><span class="fu">sudo</span> vim /etc/apache2/sites-available/000-default.conf</span>
<span id="cb35-14"><a href="web-server-manual.html#cb35-14" tabindex="-1"></a></span>
<span id="cb35-15"><a href="web-server-manual.html#cb35-15" tabindex="-1"></a><span class="op">&lt;</span>VirtualHost <span class="ex">*:80</span><span class="op">&gt;</span></span>
<span id="cb35-16"><a href="web-server-manual.html#cb35-16" tabindex="-1"></a><span class="co"># The ServerName directive sets the request scheme, hostname and port that the server uses to identify itself. This is used when creating redirection URLs. In the context of virtual hosts, the ServerName specifies what hostname must appear in the request&#39;s Host: header to match this virtual host. For the default virtual host (this file) this value is not decisive as it is used as a last resort host regard less. However, you must set it for any further virtual host explicitly.</span></span>
<span id="cb35-17"><a href="web-server-manual.html#cb35-17" tabindex="-1"></a></span>
<span id="cb35-18"><a href="web-server-manual.html#cb35-18" tabindex="-1"></a><span class="ex">ServerName</span> www.mnet4all.com</span>
<span id="cb35-19"><a href="web-server-manual.html#cb35-19" tabindex="-1"></a><span class="ex">ServerAdmin</span> webmaster@localhost</span>
<span id="cb35-20"><a href="web-server-manual.html#cb35-20" tabindex="-1"></a><span class="ex">DocumentRoot</span> /var/www/html</span>
<span id="cb35-21"><a href="web-server-manual.html#cb35-21" tabindex="-1"></a></span>
<span id="cb35-22"><a href="web-server-manual.html#cb35-22" tabindex="-1"></a><span class="co"># Available loglevels: trace8, ..., tracel, debug, info, notice, warn, error,  crit, alert, emerg. It is also possible to configure the loglevel for particular modules, e.g.</span></span>
<span id="cb35-23"><a href="web-server-manual.html#cb35-23" tabindex="-1"></a><span class="co"># LogLevel info ssl:warn</span></span>
<span id="cb35-24"><a href="web-server-manual.html#cb35-24" tabindex="-1"></a></span>
<span id="cb35-25"><a href="web-server-manual.html#cb35-25" tabindex="-1"></a><span class="ex">ErrorLog</span> <span class="va">${APACHE_LOG_DIR}</span>/error.log</span>
<span id="cb35-26"><a href="web-server-manual.html#cb35-26" tabindex="-1"></a><span class="ex">CustomLog</span> <span class="va">${APACHE_LOG_DIR}</span>/access.log combined</span>
<span id="cb35-27"><a href="web-server-manual.html#cb35-27" tabindex="-1"></a></span>
<span id="cb35-28"><a href="web-server-manual.html#cb35-28" tabindex="-1"></a><span class="co"># For most configuration files from conf-available/, which are enabled or disabled at a global level, it is possible to include a line for only one particular virtual host. For example the following line enables the CGI configuration for this host only after it has been globally disabled with &quot;a2disconf&quot;. Include conf-available/serve-cgi-bin.conf</span></span>
<span id="cb35-29"><a href="web-server-manual.html#cb35-29" tabindex="-1"></a></span>
<span id="cb35-30"><a href="web-server-manual.html#cb35-30" tabindex="-1"></a><span class="ex">ProxyPass</span> /MNet http://localhost:3838/MNet</span>
<span id="cb35-31"><a href="web-server-manual.html#cb35-31" tabindex="-1"></a><span class="ex">ProxyPassReverse</span> /MNet http://localhost: 3838/MNet</span>
<span id="cb35-32"><a href="web-server-manual.html#cb35-32" tabindex="-1"></a></span>
<span id="cb35-33"><a href="web-server-manual.html#cb35-33" tabindex="-1"></a><span class="op">&lt;</span>Location <span class="ex">/MNet</span><span class="op">&gt;</span></span>
<span id="cb35-34"><a href="web-server-manual.html#cb35-34" tabindex="-1"></a>  <span class="ex">ProxyPassReverse</span> / </span>
<span id="cb35-35"><a href="web-server-manual.html#cb35-35" tabindex="-1"></a>  <span class="ex">Order</span> allow,deny</span>
<span id="cb35-36"><a href="web-server-manual.html#cb35-36" tabindex="-1"></a>  <span class="ex">Allow</span> from all</span>
<span id="cb35-37"><a href="web-server-manual.html#cb35-37" tabindex="-1"></a><span class="op">&lt;</span>/Location<span class="op">&gt;</span></span>
<span id="cb35-38"><a href="web-server-manual.html#cb35-38" tabindex="-1"></a></span>
<span id="cb35-39"><a href="web-server-manual.html#cb35-39" tabindex="-1"></a><span class="op">&lt;</span>/VirtualHost<span class="op">&gt;</span></span>
<span id="cb35-40"><a href="web-server-manual.html#cb35-40" tabindex="-1"></a>    </span>
<span id="cb35-41"><a href="web-server-manual.html#cb35-41" tabindex="-1"></a><span class="co"># vim: syntax=apache ts=4 sw=4 sts=4 sr noet</span></span>
<span id="cb35-42"><a href="web-server-manual.html#cb35-42" tabindex="-1"></a>  </span>
<span id="cb35-43"><a href="web-server-manual.html#cb35-43" tabindex="-1"></a><span class="co">## 4. Static Sites</span></span>
<span id="cb35-44"><a href="web-server-manual.html#cb35-44" tabindex="-1"></a><span class="fu">cp</span> <span class="at">-r</span> ~/MNet/pkgdown/docs /var/www/html/mnet-docs</span>
<span id="cb35-45"><a href="web-server-manual.html#cb35-45" tabindex="-1"></a><span class="fu">cp</span> <span class="at">-r</span> ~/MNet/bookdown/book /var/www/html/mnet-manual</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="applying-mnet-in-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/01-WebServer.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
